--- a/net/minecraft/world/level/Level.java
+++ b/net/minecraft/world/level/Level.java
@@ -131,7 +131,7 @@
         this.dimension = p_270683_;
         this.isClientSide = p_270904_;
         if (dimensiontype.coordinateScale() != 1.0) {
-            this.worldBorder = new WorldBorder(this) {
+            this.worldBorder = new WorldBorder() {
                 @Override
                 public double getCenterX() {
                     return super.getCenterX() / dimensiontype.coordinateScale();
