--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -497,7 +497,7 @@
 
         this.firstTick = false;
         if (this.level() instanceof ServerLevel serverlevel1 && this instanceof Leashable) {
-            Leashable.tickLeash(serverlevel1, (Entity)((Leashable)this));
+            Leashable.tickLeash(serverlevel1, (Entity & Leashable)this);
         }
 
         profilerfiller.pop();
