--- a/net/minecraft/world/timeline/Timeline.java
+++ b/net/minecraft/world/timeline/Timeline.java
@@ -32,7 +_,7 @@
 
     public static final Codec<Holder<Timeline>> CODEC = RegistryFixedCodec.create(Registries.TIMELINE);
     private static final Codec<Map<EnvironmentAttribute<?>, AttributeTrack<?, ?>>> TRACKS_CODEC = Codec.dispatchedMap(EnvironmentAttributes.CODEC, Util.memoize(AttributeTrack::createCodec));
-    public static final Codec<Timeline> DIRECT_CODEC = RecordCodecBuilder.create(i -> i.group(WorldClock.CODEC.fieldOf("clock").forGetter(t -> t.clock), ExtraCodecs.POSITIVE_INT.optionalFieldOf("period_ticks").forGetter(t -> t.periodTicks), TRACKS_CODEC.optionalFieldOf("tracks", Map.of()).forGetter(t -> t.tracks), Codec.unboundedMap(ClockTimeMarker.KEY_CODEC, TimeMarkerInfo.CODEC).optionalFieldOf("time_markers", Map.of()).forGetter(t -> t.timeMarkers)).apply(i, Timeline::new)).validate(Timeline::validateInternal);
+    public static final Codec<Timeline> DIRECT_CODEC = RecordCodecBuilder.<Timeline>create(i -> i.group(WorldClock.CODEC.fieldOf("clock").forGetter(t -> t.clock), ExtraCodecs.POSITIVE_INT.optionalFieldOf("period_ticks").forGetter(t -> t.periodTicks), TRACKS_CODEC.optionalFieldOf("tracks", Map.of()).forGetter(t -> t.tracks), Codec.unboundedMap(ClockTimeMarker.KEY_CODEC, TimeMarkerInfo.CODEC).optionalFieldOf("time_markers", Map.of()).forGetter(t -> t.timeMarkers)).apply(i, Timeline::new)).validate(Timeline::validateInternal);
     public static final Codec<Timeline> NETWORK_CODEC = DIRECT_CODEC.xmap(Timeline::filterSyncableTracks, Timeline::filterSyncableTracks);
     private final Holder<WorldClock> clock;
     private final Optional<Integer> periodTicks;
