--- a/net/minecraft/commands/arguments/item/FunctionArgument.java
+++ b/net/minecraft/commands/arguments/item/FunctionArgument.java
@@ -32,7 +32,7 @@
         if (p_120909_.canRead() && p_120909_.peek() == '#') {
             p_120909_.skip();
             final ResourceLocation resourcelocation1 = ResourceLocation.read(p_120909_);
-            return new FunctionArgument.Result(this) {
+            return new FunctionArgument.Result() {
                 @Override
                 public Collection<CommandFunction<CommandSourceStack>> create(CommandContext<CommandSourceStack> p_120943_) throws CommandSyntaxException {
                     return FunctionArgument.getFunctionTag(p_120943_, resourcelocation1);
@@ -52,7 +52,7 @@
             };
         } else {
             final ResourceLocation resourcelocation = ResourceLocation.read(p_120909_);
-            return new FunctionArgument.Result(this) {
+            return new FunctionArgument.Result() {
                 @Override
                 public Collection<CommandFunction<CommandSourceStack>> create(CommandContext<CommandSourceStack> p_120952_) throws CommandSyntaxException {
                     return Collections.singleton(FunctionArgument.getFunction(p_120952_, resourcelocation));
