--- a/net/minecraft/commands/arguments/item/ItemParser.java
+++ b/net/minecraft/commands/arguments/item/ItemParser.java
@@ -64,7 +64,7 @@
     public ItemParser.ItemResult parse(StringReader p_324270_) throws CommandSyntaxException {
         final MutableObject<Holder<Item>> mutableobject = new MutableObject<>();
         final DataComponentMap.Builder datacomponentmap$builder = DataComponentMap.builder();
-        this.parse(p_324270_, new ItemParser.Visitor(this) {
+        this.parse(p_324270_, new ItemParser.Visitor() {
             @Override
             public void visitItem(Holder<Item> p_324335_) {
                 mutableobject.setValue(p_324335_);
