--- a/net/minecraft/world/entity/ai/behavior/ReactToBell.java
+++ b/net/minecraft/world/entity/ai/behavior/ReactToBell.java
@@ -11,7 +11,7 @@
 public class ReactToBell {
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            p_259349_ -> p_259349_.<MemoryAccessor>group(p_259349_.present(MemoryModuleType.HEARD_BELL_TIME))
+            p_259349_ -> p_259349_.group(p_259349_.present(MemoryModuleType.HEARD_BELL_TIME))
                     .apply(p_259349_, p_259472_ -> (p_340011_, p_340012_, p_340013_) -> {
                             Raid raid = p_340011_.getRaidAt(p_340012_.blockPosition());
                             if (raid == null) {
