--- a/net/minecraft/world/flag/FeatureFlags.java
+++ b/net/minecraft/world/flag/FeatureFlags.java
@@ -11,9 +11,9 @@
     public static final FeatureFlag UPDATE_1_21;
     public static final FeatureFlag TRADE_REBALANCE;
     public static final FeatureFlagRegistry REGISTRY;
-    public static final Codec<FeatureFlagSet> CODEC = REGISTRY.codec();
-    public static final FeatureFlagSet VANILLA_SET = FeatureFlagSet.of(VANILLA);
-    public static final FeatureFlagSet DEFAULT_FLAGS = VANILLA_SET;
+    public static final Codec<FeatureFlagSet> CODEC;
+    public static final FeatureFlagSet VANILLA_SET;
+    public static final FeatureFlagSet DEFAULT_FLAGS;
 
     public static String printMissingFlags(FeatureFlagSet p_250581_, FeatureFlagSet p_250326_) {
         return printMissingFlags(REGISTRY, p_250581_, p_250326_);
@@ -36,5 +36,8 @@
         TRADE_REBALANCE = featureflagregistry$builder.createVanilla("trade_rebalance");
         UPDATE_1_21 = featureflagregistry$builder.createVanilla("update_1_21");
         REGISTRY = featureflagregistry$builder.build();
+        CODEC = REGISTRY.codec();
+        VANILLA_SET = FeatureFlagSet.of(VANILLA);
+        DEFAULT_FLAGS = VANILLA_SET;
     }
 }
