--- a/net/minecraft/network/protocol/game/ServerboundConfigurationAcknowledgedPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundConfigurationAcknowledgedPacket.java
@@ -4,14 +4,11 @@
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.protocol.Packet;
 
-public final class ServerboundConfigurationAcknowledgedPacket extends Record implements Packet<ServerGamePacketListener> {
+public record ServerboundConfigurationAcknowledgedPacket() implements Packet<ServerGamePacketListener> {
    public ServerboundConfigurationAcknowledgedPacket(FriendlyByteBuf p_294397_) {
       this();
    }
 
-   public ServerboundConfigurationAcknowledgedPacket() {
-   }
-
    public void m_5779_(FriendlyByteBuf p_294932_) {
    }
 
@@ -22,16 +19,4 @@
    public ConnectionProtocol m_291754_() {
       return ConnectionProtocol.CONFIGURATION;
    }
-
-   public final String toString() {
-      return this.toString<invokedynamic>(this);
    }
-
-   public final int hashCode() {
-      return this.hashCode<invokedynamic>(this);
-   }
-
-   public final boolean equals(Object p_295779_) {
-      return this.equals<invokedynamic>(this, p_295779_);
-   }
-}
