--- a/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
@@ -32,7 +32,7 @@
     private static DataResult<LootItemBlockStatePropertyCondition> validate(LootItemBlockStatePropertyCondition p_298471_) {
         return p_298471_.properties()
             .flatMap(p_298822_ -> p_298822_.checkState(p_298471_.block().value().getStateDefinition()))
-            .map(p_299129_ -> DataResult.error(() -> "Block " + p_298471_.block() + " has no property" + p_299129_))
+            .map(p_299129_ -> DataResult.<LootItemBlockStatePropertyCondition>error(() -> "Block " + p_298471_.block() + " has no property" + p_299129_))
             .orElse(DataResult.success(p_298471_));
     }
 
