--- a/net/minecraft/client/gui/screens/worldselection/WorldCreationUiState.java
+++ b/net/minecraft/client/gui/screens/worldselection/WorldCreationUiState.java
@@ -315,7 +315,7 @@
         public Component describePreset() {
             return Optional.ofNullable(this.preset)
                 .flatMap(Holder::unwrapKey)
-                .map(p_465546_ -> Component.translatable(p_465546_.identifier().toLanguageKey("generator")))
+                .map(p_465546_ -> (Component) Component.translatable(p_465546_.identifier().toLanguageKey("generator")))
                 .orElse(CUSTOM_WORLD_DESCRIPTION);
         }
 
