--- a/net/minecraft/client/gui/screens/worldselection/WorldCreationUiState.java
+++ b/net/minecraft/client/gui/screens/worldselection/WorldCreationUiState.java
@@ -303,7 +303,7 @@
         public Component describePreset() {
             return Optional.ofNullable(this.preset)
                 .flatMap(Holder::unwrapKey)
-                .map(p_268048_ -> Component.translatable(p_268048_.location().toLanguageKey("generator")))
+                .map(p_268048_ -> (Component) Component.translatable(p_268048_.location().toLanguageKey("generator")))
                 .orElse(CUSTOM_WORLD_DESCRIPTION);
         }
 
