--- a/net/minecraft/client/telemetry/TelemetryPropertyMap.java
+++ b/net/minecraft/client/telemetry/TelemetryPropertyMap.java
@@ -61,7 +61,7 @@
                     T t = p_261668_.get(p_261627_.id());
                     if (t != null) {
                         DataResult<V> dataresult = p_261627_.codec().parse(p_261859_, t);
-                        return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.put(p_261627_, (T)p_261796_), dataresult);
+                        return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.put((TelemetryProperty<T>)p_261627_, (T)p_261796_), dataresult);
                     } else {
                         return p_261892_;
                     }
