--- a/com/mojang/blaze3d/buffers/GpuBuffer.java
+++ b/com/mojang/blaze3d/buffers/GpuBuffer.java
@@ -70,6 +70,7 @@
     @Retention(RetentionPolicy.CLASS)
     @Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.METHOD, ElementType.TYPE_USE})
     @OnlyIn(Dist.CLIENT)
+    @org.intellij.lang.annotations.MagicConstant(flagsFromClass = GpuBuffer.class)
     public @interface Usage {
     }
 }
