--- a/net/minecraft/client/multiplayer/ClientLevel.java
+++ b/net/minecraft/client/multiplayer/ClientLevel.java
@@ -1116,6 +1116,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.addAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
@@ -1129,6 +1130,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.removeAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
