--- a/net/minecraft/data/registries/VanillaRegistries.java
+++ b/net/minecraft/data/registries/VanillaRegistries.java
@@ -38,8 +38,8 @@
 public class VanillaRegistries {
     private static final RegistrySetBuilder BUILDER = new RegistrySetBuilder()
         .add(Registries.DIMENSION_TYPE, DimensionTypes::bootstrap)
-        .add(Registries.CONFIGURED_CARVER, Carvers::bootstrap)
-        .add(Registries.CONFIGURED_FEATURE, FeatureUtils::bootstrap)
+        .add(Registries.CONFIGURED_CARVER, (RegistrySetBuilder.RegistryBootstrap)Carvers::bootstrap)
+        .add(Registries.CONFIGURED_FEATURE, (RegistrySetBuilder.RegistryBootstrap)FeatureUtils::bootstrap)
         .add(Registries.PLACED_FEATURE, PlacementUtils::bootstrap)
         .add(Registries.STRUCTURE, Structures::bootstrap)
         .add(Registries.STRUCTURE_SET, StructureSets::bootstrap)
