--- a/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
@@ -143,8 +143,8 @@
             p_16633_ -> p_16633_.update(
                     DSL.remainderFinder(),
                     p_145575_ -> p_145575_.getMapValues().map(p_145578_ -> p_145575_.createMap(p_145578_.entrySet().stream().map(p_145580_ -> {
-                                if (((Dynamic)p_145580_.getKey()).asString("").startsWith("key_")) {
-                                    int i = Integer.parseInt(((Dynamic)p_145580_.getValue()).asString(""));
+                                if ((p_145580_.getKey()).asString("").startsWith("key_")) {
+                                    int i = Integer.parseInt((p_145580_.getValue()).asString(""));
                                     if (i < 0) {
                                         int j = i + 100;
                                         String s1;
@@ -158,15 +158,15 @@
                                             s1 = "key.mouse." + (j + 1);
                                         }
         
-                                        return Pair.of((Dynamic)p_145580_.getKey(), ((Dynamic)p_145580_.getValue()).createString(s1));
+                                        return Pair.of(p_145580_.getKey(), (p_145580_.getValue()).createString(s1));
                                     } else {
                                         String s = MAP.getOrDefault(i, "key.unknown");
-                                        return Pair.of((Dynamic)p_145580_.getKey(), ((Dynamic)p_145580_.getValue()).createString(s));
+                                        return Pair.of(p_145580_.getKey(), (p_145580_.getValue()).createString(s));
                                     }
                                 } else {
-                                    return Pair.of((Dynamic)p_145580_.getKey(), (Dynamic)p_145580_.getValue());
+                                    return Pair.of(p_145580_.getKey(), p_145580_.getValue());
                                 }
-                            }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse(p_145575_)
+                            }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse((Dynamic)p_145575_)
                 )
         );
     }
