--- a/net/minecraft/util/AbstractListBuilder.java
+++ b/net/minecraft/util/AbstractListBuilder.java
@@ -27,7 +27,7 @@
 
     @Override
     public ListBuilder<T> add(T p_412702_) {
-        this.builder = this.builder.map(p_412090_ -> (T)this.append((B)p_412090_, p_412702_));
+        this.builder = this.builder.map(p_412090_ -> this.append((B)p_412090_, p_412702_));
         return this;
     }
 
