--- a/net/minecraft/server/jsonrpc/methods/DiscoveryService.java
+++ b/net/minecraft/server/jsonrpc/methods/DiscoveryService.java
@@ -43,7 +43,7 @@
         private static MapCodec<DiscoveryService.DiscoverComponents> typedSchema() {
             return RecordCodecBuilder.mapCodec(
                 p_451637_ -> p_451637_.group(
-                        Codec.unboundedMap(Codec.STRING, Schema.CODEC).fieldOf("schemas").forGetter(DiscoveryService.DiscoverComponents::schemas)
+                        Codec.unboundedMap(Codec.STRING, (Codec<Schema<?>>) Schema.CODEC).fieldOf("schemas").forGetter(DiscoveryService.DiscoverComponents::schemas)
                     )
                     .apply(p_451637_, DiscoveryService.DiscoverComponents::new)
             );
