--- a/com/mojang/realmsclient/RealmsMainScreen.java
+++ b/com/mojang/realmsclient/RealmsMainScreen.java
@@ -204,8 +204,7 @@
 
          this.layout = this.createLayout(p_294474_);
          this.activeLayoutState = p_294474_;
-         this.layout.visitWidgets(p_272289_ -> {
-         });
+         this.layout.visitWidgets(this::addRenderableWidget);
          this.repositionElements();
       }
    }
