--- a/com/mojang/blaze3d/platform/InputConstants.java
+++ b/com/mojang/blaze3d/platform/InputConstants.java
@@ -499,6 +499,7 @@
     @Retention(RetentionPolicy.CLASS)
     @Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.METHOD, ElementType.TYPE_USE})
     @OnlyIn(Dist.CLIENT)
+    @org.intellij.lang.annotations.MagicConstant(valuesFromClass = InputConstants.class)
     public @interface Value {
     }
 }
