--- a/com/mojang/realmsclient/gui/screens/RealmsResetNormalWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsResetNormalWorldScreen.java
@@ -61,8 +61,7 @@
         LinearLayout linearlayout1 = this.layout.addToFooter(LinearLayout.horizontal().spacing(10));
         linearlayout1.addChild(Button.builder(this.buttonTitle, p_293585_ -> this.callback.accept(this.createWorldGenerationInfo())).build());
         linearlayout1.addChild(Button.builder(CommonComponents.GUI_BACK, p_89288_ -> this.onClose()).build());
-        this.layout.visitWidgets(p_321344_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
