--- a/net/minecraft/world/level/storage/loot/entries/TagEntry.java
+++ b/net/minecraft/world/level/storage/loot/entries/TagEntry.java
@@ -51,7 +51,7 @@
             return false;
         } else {
             for(final Holder<Item> holder : BuiltInRegistries.ITEM.getTagOrEmpty(this.tag)) {
-                p_79847_.accept(new LootPoolSingletonContainer.EntryBase(this) {
+                p_79847_.accept(new LootPoolSingletonContainer.EntryBase() {
                     @Override
                     public void createItemStack(Consumer<ItemStack> p_79869_, LootContext p_79870_) {
                         p_79869_.accept(new ItemStack(holder));
