--- a/net/minecraft/advancements/criterion/MinMaxBounds.java
+++ b/net/minecraft/advancements/criterion/MinMaxBounds.java
@@ -174,7 +_,7 @@
         }
 
         public static <T extends Number & Comparable<T>> Bounds<T> any() {
-            return new Bounds<>(Optional.empty(), Optional.empty());
+            return new Bounds<>(Optional.<T>empty(), Optional.<T>empty());
         }
 
         public static <T extends Number & Comparable<T>> Bounds<T> exactly(final T value) {
