--- a/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
@@ -54,8 +54,7 @@
             Button.builder(CommonComponents.GUI_DONE, p_101059_ -> this.exitCallback.accept(Optional.of(this.gameRules))).build()
         );
         gridlayout$rowhelper.addChild(Button.builder(CommonComponents.GUI_CANCEL, p_101073_ -> this.exitCallback.accept(Optional.empty())).build());
-        gridlayout$rowhelper.getGrid().visitWidgets(p_267855_ -> {
-        });
+        gridlayout$rowhelper.getGrid().visitWidgets(this::addRenderableWidget);
         gridlayout$rowhelper.getGrid().setPosition(this.width / 2 - 155, this.height - 28);
         gridlayout$rowhelper.getGrid().arrangeElements();
     }
