--- a/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
@@ -116,8 +116,7 @@
         );
         this.doneButton = linearlayout1.addChild(Button.builder(CommonComponents.GUI_DONE, p_100036_ -> this.onClose()).build());
         this.reload();
-        this.layout.visitWidgets(p_329736_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
@@ -221,7 +220,7 @@
                         if (p_293606_) {
                             List<Path> list = new ArrayList<>(p_100029_.size());
                             Set<Path> set = new HashSet<>(p_100029_);
-                            PackDetector<Path> packdetector = new PackDetector<Path>(this, this.minecraft.directoryValidator()) {
+                            PackDetector<Path> packdetector = new PackDetector<Path>(this.minecraft.directoryValidator()) {
                                 protected Path createZipPack(Path p_294508_) {
                                     return p_294508_;
                                 }
