--- a/net/minecraft/world/item/crafting/RecipeManager.java
+++ b/net/minecraft/world/item/crafting/RecipeManager.java
@@ -110,7 +110,7 @@
     }
 
     private <I extends RecipeInput, T extends Recipe<I>> Collection<RecipeHolder<T>> byType(RecipeType<T> p_44055_) {
-        return (Collection<RecipeHolder<T>>)this.byType.get(p_44055_);
+        return (Collection<RecipeHolder<T>>)(Collection<?>)this.byType.get(p_44055_);
     }
 
     public <I extends RecipeInput, T extends Recipe<I>> NonNullList<ItemStack> getRemainingItemsFor(RecipeType<T> p_44070_, I p_345118_, Level p_44072_) {
