--- a/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
@@ -59,8 +59,7 @@
         LinearLayout linearlayout1 = this.layout.addToFooter(LinearLayout.horizontal().spacing(10));
         linearlayout1.addChild(button);
         linearlayout1.addChild(Button.builder(CommonComponents.GUI_BACK, p_293570_ -> this.onClose()).build());
-        this.layout.visitWidgets(p_293566_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
         this.setInitialFocus(this.nameBox);
     }
