--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -146,7 +146,7 @@
                 break;
             case DIRECT:
                 this.writeVarInt(0);
-                p_263358_.accept((T)this, p_263384_.value());
+                p_263358_.accept(this, p_263384_.value());
         }
     }
 
@@ -159,7 +159,7 @@
     public <T> Holder<T> readById(IdMap<Holder<T>> p_263401_, FriendlyByteBuf.Reader<T> p_263374_) {
         int i = this.readVarInt();
         if (i == 0) {
-            return Holder.direct(p_263374_.apply((T)this));
+            return Holder.direct(p_263374_.apply(this));
         } else {
             Holder<T> holder = p_263401_.byId(i - 1);
             if (holder == null) {
@@ -185,7 +185,7 @@
         C c = p_236839_.apply(i);
 
         for(int j = 0; j < i; ++j) {
-            c.add(p_236840_.apply((T)this));
+            c.add(p_236840_.apply(this));
         }
 
         return c;
@@ -195,7 +195,7 @@
         this.writeVarInt(p_236829_.size());
 
         for(T t : p_236829_) {
-            p_236830_.accept((T)this, t);
+            p_236830_.accept(this, t);
         }
     }
 
@@ -224,8 +224,8 @@
         M m = p_236842_.apply(i);
 
         for(int j = 0; j < i; ++j) {
-            K k = p_236843_.apply((K)this);
-            V v = p_236844_.apply((V)this);
+            K k = p_236843_.apply(this);
+            V v = p_236844_.apply(this);
             m.put(k, v);
         }
 
@@ -239,8 +239,8 @@
     public <K, V> void writeMap(Map<K, V> p_236832_, FriendlyByteBuf.Writer<K> p_236833_, FriendlyByteBuf.Writer<V> p_236834_) {
         this.writeVarInt(p_236832_.size());
         p_236832_.forEach((p_236856_, p_236857_) -> {
-            p_236833_.accept((K)this, p_236856_);
-            p_236834_.accept((V)this, p_236857_);
+            p_236833_.accept(this, p_236856_);
+            p_236834_.accept(this, p_236857_);
         });
     }
 
@@ -280,25 +280,25 @@
     public <T> void writeOptional(Optional<T> p_236836_, FriendlyByteBuf.Writer<T> p_236837_) {
         if (p_236836_.isPresent()) {
             this.writeBoolean(true);
-            p_236837_.accept((T)this, p_236836_.get());
+            p_236837_.accept(this, p_236836_.get());
         } else {
             this.writeBoolean(false);
         }
     }
 
     public <T> Optional<T> readOptional(FriendlyByteBuf.Reader<T> p_236861_) {
-        return this.readBoolean() ? Optional.of(p_236861_.apply((T)this)) : Optional.empty();
+        return this.readBoolean() ? Optional.of(p_236861_.apply(this)) : Optional.empty();
     }
 
     @Nullable
     public <T> T readNullable(FriendlyByteBuf.Reader<T> p_236869_) {
-        return this.readBoolean() ? p_236869_.apply((T)this) : null;
+        return this.readBoolean() ? p_236869_.apply(this) : null;
     }
 
     public <T> void writeNullable(@Nullable T p_236822_, FriendlyByteBuf.Writer<T> p_236823_) {
         if (p_236822_ != null) {
             this.writeBoolean(true);
-            p_236823_.accept((T)this, p_236822_);
+            p_236823_.accept(this, p_236822_);
         } else {
             this.writeBoolean(false);
         }
@@ -307,15 +307,15 @@
     public <L, R> void writeEither(Either<L, R> p_236811_, FriendlyByteBuf.Writer<L> p_236812_, FriendlyByteBuf.Writer<R> p_236813_) {
         p_236811_.ifLeft(p_293716_ -> {
             this.writeBoolean(true);
-            p_236812_.accept((L)this, p_293716_);
+            p_236812_.accept(this, p_293716_);
         }).ifRight(p_293718_ -> {
             this.writeBoolean(false);
-            p_236813_.accept((R)this, p_293718_);
+            p_236813_.accept(this, p_293718_);
         });
     }
 
     public <L, R> Either<L, R> readEither(FriendlyByteBuf.Reader<L> p_236863_, FriendlyByteBuf.Reader<R> p_236864_) {
-        return this.readBoolean() ? Either.left(p_236863_.apply((L)this)) : Either.right(p_236864_.apply((R)this));
+        return this.readBoolean() ? Either.left(p_236863_.apply(this)) : Either.right(p_236864_.apply(this));
     }
 
     public byte[] readByteArray() {
