--- a/net/minecraft/TracingExecutor.java
+++ b/net/minecraft/TracingExecutor.java
@@ -21,7 +21,7 @@
                     }
                 });
         } else {
-            return (Executor)(TracyClient.isAvailable() ? p_372837_ -> this.service.execute(() -> {
+            return (Executor)(TracyClient.isAvailable() ? (Executor) p_372837_ -> this.service.execute(() -> {
                     try (Zone zone = TracyClient.beginZone(p_372983_, SharedConstants.IS_RUNNING_IN_IDE)) {
                         p_372837_.run();
                     }
