--- a/net/minecraft/world/entity/TamableAnimal.java
+++ b/net/minecraft/world/entity/TamableAnimal.java
@@ -294,7 +294,7 @@
     }
 
     public class TamableAnimalPanicGoal extends PanicGoal {
-        public TamableAnimalPanicGoal(double param3, TagKey<DamageType> TamableAnimal.this) {
+        public TamableAnimalPanicGoal(double p_350422_, TagKey<DamageType> p_350418_) {
             super(TamableAnimal.this, p_350422_, p_350418_);
         }
 
