--- a/net/minecraft/world/level/storage/loot/functions/SetCustomModelDataFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetCustomModelDataFunction.java
@@ -22,7 +_,7 @@
 
 public class SetCustomModelDataFunction extends LootItemConditionalFunction {
     private static final Codec<Holder<ContextIntProvider>> COLOR_PROVIDER_CODEC = Codec.withAlternative(
-        ContextIntProviders.CODEC, ExtraCodecs.RGB_COLOR_CODEC, ContextIntProviders::exactly
+        ContextIntProviders.CODEC, ExtraCodecs.RGB_COLOR_CODEC, (Function<Integer, Holder<ContextIntProvider>>)ContextIntProviders::exactly
     );
     public static final MapCodec<SetCustomModelDataFunction> MAP_CODEC = RecordCodecBuilder.mapCodec(
         i -> commonFields(i)
