--- a/net/minecraft/server/dedicated/ServerWatchdog.java
+++ b/net/minecraft/server/dedicated/ServerWatchdog.java
@@ -96,7 +96,7 @@
     private void exit() {
         try {
             Timer timer = new Timer();
-            timer.schedule(new TimerTask(this) {
+            timer.schedule(new TimerTask() {
                 @Override
                 public void run() {
                     Runtime.getRuntime().halt(1);
