--- a/net/minecraft/client/main/Main.java
+++ b/net/minecraft/client/main/Main.java
@@ -202,7 +202,7 @@
         };
         thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(logger));
         Runtime.getRuntime().addShutdownHook(thread);
-        final Minecraft minecraft = null;
+        Minecraft minecraft = null;
 
         try {
             Thread.currentThread().setName("Render thread");
@@ -230,7 +230,7 @@
                 @Override
                 public void run() {
                     RenderSystem.initGameThread(true);
-                    minecraft.run();
+                    minecraft1.run();
                 }
             };
             thread1.start();
