--- a/net/minecraft/resources/RegistryFileCodec.java
+++ b/net/minecraft/resources/RegistryFileCodec.java
@@ -68,7 +68,7 @@
                     return holdergetter.get(resourcekey)
                         .map(DataResult::success)
                         .orElseGet(() -> DataResult.error(() -> "Failed to get element " + resourcekey))
-                        .<Pair<? super Holder.Reference<E>, T>>map(p_255658_ -> Pair.of(p_255658_, pair.getSecond()))
+                        .<Pair<Holder<E>, T>>map(p_255658_ -> Pair.of(p_255658_, pair.getSecond()))
                         .setLifecycle(Lifecycle.stable());
                 }
             }
