--- a/net/minecraft/world/level/levelgen/SurfaceSystem.java
+++ b/net/minecraft/world/level/levelgen/SurfaceSystem.java
@@ -77,7 +77,7 @@
         final ChunkPos chunkpos = p_224654_.getPos();
         int i = chunkpos.getMinBlockX();
         int j = chunkpos.getMinBlockZ();
-        BlockColumn blockcolumn = new BlockColumn(this) {
+        BlockColumn blockcolumn = new BlockColumn() {
             @Override
             public BlockState getBlock(int p_190006_) {
                 return p_224654_.getBlockState(blockpos$mutableblockpos.setY(p_190006_));
