--- a/net/minecraft/server/dialog/input/TextInput.java
+++ b/net/minecraft/server/dialog/input/TextInput.java
@@ -12,7 +_,7 @@
 
 public record TextInput(int width, Component label, boolean labelVisible, String initial, int maxLength, Optional<MultilineOptions> multiline) implements InputControl {
 
-    public static final MapCodec<TextInput> MAP_CODEC = RecordCodecBuilder.mapCodec(i -> i.group(Dialog.WIDTH_CODEC.optionalFieldOf("width", Integer.valueOf(200)).forGetter(TextInput::width), ComponentSerialization.CODEC.fieldOf("label").forGetter(TextInput::label), Codec.BOOL.optionalFieldOf("label_visible", Boolean.valueOf(true)).forGetter(TextInput::labelVisible), Codec.STRING.optionalFieldOf("initial", "").forGetter(TextInput::initial), ExtraCodecs.POSITIVE_INT.optionalFieldOf("max_length", Integer.valueOf(32)).forGetter(TextInput::maxLength), MultilineOptions.CODEC.optionalFieldOf("multiline").forGetter(TextInput::multiline)).apply(i, TextInput::new)).validate(o -> {
+    public static final MapCodec<TextInput> MAP_CODEC = RecordCodecBuilder.<TextInput>mapCodec(i -> i.group(Dialog.WIDTH_CODEC.optionalFieldOf("width", Integer.valueOf(200)).forGetter(TextInput::width), ComponentSerialization.CODEC.fieldOf("label").forGetter(TextInput::label), Codec.BOOL.optionalFieldOf("label_visible", Boolean.valueOf(true)).forGetter(TextInput::labelVisible), Codec.STRING.optionalFieldOf("initial", "").forGetter(TextInput::initial), ExtraCodecs.POSITIVE_INT.optionalFieldOf("max_length", Integer.valueOf(32)).forGetter(TextInput::maxLength), MultilineOptions.CODEC.optionalFieldOf("multiline").forGetter(TextInput::multiline)).apply(i, TextInput::new)).validate(o -> {
         if (o.initial.length() > o.maxLength()) {
             return DataResult.error(() -> "Default text length exceeds allowed size");
         }
