--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -225,7 +_,7 @@
         default Function<OptionInstance<T>, AbstractWidget> createButton(
             OptionInstance.TooltipSupplier<T> tooltip, Options options, int x, int y, int width, Consumer<T> onValueChanged
         ) {
-            return instance -> CycleButton.builder(instance.toString, instance::get)
+            return instance -> CycleButton.<T>builder(instance.toString, instance::get)
                 .withValues(this.valueListSupplier())
                 .withTooltip(tooltip)
                 .create(x, y, width, 20, instance.caption, (button, value) -> {
