--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -185,7 +_,7 @@
         }
 
         default Function<OptionInstance<T>, AbstractWidget> createButton(final TooltipSupplier<T> tooltip, final Options options, final int x, final int y, final int width, final Consumer<T> onValueChanged) {
-            return instance -> CycleButton.builder(instance.toString, instance::get).withValues(valueListSupplier()).withTooltip(tooltip).create(x, y, width, 20, instance.caption, (button, value) -> {
+            return instance -> CycleButton.builder(instance.toString, (Supplier<T>) instance::get).withValues(valueListSupplier()).withTooltip(tooltip).create(x, y, width, 20, instance.caption, (button, value) -> {
                 valueSetter().set(instance, value);
                 options.save();
                 onValueChanged.accept(value);
