--- a/net/minecraft/world/inventory/ItemCombinerMenu.java
+++ b/net/minecraft/world/inventory/ItemCombinerMenu.java
@@ -44,7 +44,6 @@
         for(final ItemCombinerMenuSlotDefinition.SlotDefinition itemcombinermenuslotdefinition$slotdefinition : p_267172_.getSlots()) {
             this.addSlot(
                 new Slot(
-                    this,
                     this.inputSlots,
                     itemcombinermenuslotdefinition$slotdefinition.slotIndex(),
                     itemcombinermenuslotdefinition$slotdefinition.x(),
