--- a/net/minecraft/world/inventory/LoomMenu.java
+++ b/net/minecraft/world/inventory/LoomMenu.java
@@ -65,19 +65,19 @@
     public LoomMenu(int p_39859_, Inventory p_39860_, final ContainerLevelAccess p_39861_) {
         super(MenuType.LOOM, p_39859_);
         this.access = p_39861_;
-        this.bannerSlot = this.addSlot(new Slot(this, this.inputContainer, 0, 13, 26) {
+        this.bannerSlot = this.addSlot(new Slot(this.inputContainer, 0, 13, 26) {
             @Override
             public boolean mayPlace(ItemStack p_39918_) {
                 return p_39918_.getItem() instanceof BannerItem;
             }
         });
-        this.dyeSlot = this.addSlot(new Slot(this, this.inputContainer, 1, 33, 26) {
+        this.dyeSlot = this.addSlot(new Slot(this.inputContainer, 1, 33, 26) {
             @Override
             public boolean mayPlace(ItemStack p_39927_) {
                 return p_39927_.getItem() instanceof DyeItem;
             }
         });
-        this.patternSlot = this.addSlot(new Slot(this, this.inputContainer, 2, 23, 45) {
+        this.patternSlot = this.addSlot(new Slot(this.inputContainer, 2, 23, 45) {
             @Override
             public boolean mayPlace(ItemStack p_39936_) {
                 return p_39936_.getItem() instanceof BannerPatternItem;
