--- a/net/minecraft/world/scores/PlayerTeam.java
+++ b/net/minecraft/world/scores/PlayerTeam.java
@@ -204,6 +_,7 @@
 
     @Retention(RetentionPolicy.CLASS)
     @Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.METHOD, ElementType.TYPE_USE})
+    @org.intellij.lang.annotations.MagicConstant(flags = {BIT_FRIENDLY_FIRE, BIT_SEE_INVISIBLES})
     public @interface OptionFlags {
     }
 
