--- a/com/mojang/realmsclient/RealmsMainScreen.java
+++ b/com/mojang/realmsclient/RealmsMainScreen.java
@@ -552,8 +552,7 @@
 
     @Nullable
     private RealmsServer getSelectedServer() {
-        AbstractSelectionList.Entry abstractselectionlist$entry = this.realmSelectionList.getSelected();
-        return abstractselectionlist$entry instanceof RealmsMainScreen.ServerEntry realmsmainscreen$serverentry
+        return this.realmSelectionList.getSelected() instanceof RealmsMainScreen.ServerEntry realmsmainscreen$serverentry
             ? realmsmainscreen$serverentry.getServer()
             : null;
     }
