--- a/net/minecraft/commands/execution/tasks/FallthroughTask.java
+++ b/net/minecraft/commands/execution/tasks/FallthroughTask.java
@@ -9,7 +9,7 @@
     private static final FallthroughTask<? extends ExecutionCommandSource<?>> INSTANCE = new FallthroughTask<>();
 
     public static <T extends ExecutionCommandSource<T>> EntryAction<T> instance() {
-        return INSTANCE;
+        return (EntryAction<T>)INSTANCE;
     }
 
     @Override
