--- a/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
+++ b/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
@@ -145,7 +145,7 @@
             return p_259751_ -> {
                 final BehaviorBuilder.TriggerWithResult<E, A> triggerwithresult = BehaviorBuilder.get(p_259751_);
                 final BehaviorBuilder.TriggerWithResult<E, Function<A, R>> triggerwithresult1 = BehaviorBuilder.get(p_259294_);
-                return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>(this) {
+                return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>() {
                     @Override
                     public R tryTrigger(ServerLevel p_259603_, E p_260233_, long p_259654_) {
                         A a = (A)triggerwithresult.tryTrigger(p_259603_, p_260233_, p_259654_);
@@ -172,7 +172,7 @@
 
         public <T, R> BehaviorBuilder<E, R> map(final Function<? super T, ? extends R> p_259963_, App<BehaviorBuilder.Mu<E>, T> p_260355_) {
             final BehaviorBuilder.TriggerWithResult<E, T> triggerwithresult = BehaviorBuilder.get(p_260355_);
-            return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>(this) {
+            return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>() {
                 @Override
                 public R tryTrigger(ServerLevel p_259755_, E p_259656_, long p_259300_) {
                     T t = triggerwithresult.tryTrigger(p_259755_, p_259656_, p_259300_);
@@ -197,7 +197,7 @@
             final BehaviorBuilder.TriggerWithResult<E, A> triggerwithresult = BehaviorBuilder.get(p_259162_);
             final BehaviorBuilder.TriggerWithResult<E, B> triggerwithresult1 = BehaviorBuilder.get(p_259733_);
             final BehaviorBuilder.TriggerWithResult<E, BiFunction<A, B, R>> triggerwithresult2 = BehaviorBuilder.get(p_259535_);
-            return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>(this) {
+            return BehaviorBuilder.create(new BehaviorBuilder.TriggerWithResult<E, R>() {
                 @Override
                 public R tryTrigger(ServerLevel p_259274_, E p_259817_, long p_259820_) {
                     A a = triggerwithresult.tryTrigger(p_259274_, p_259817_, p_259820_);
@@ -237,7 +237,7 @@
             final BehaviorBuilder.TriggerWithResult<E, T3> triggerwithresult2 = BehaviorBuilder.get(p_259969_);
             final BehaviorBuilder.TriggerWithResult<E, Function3<T1, T2, T3, R>> triggerwithresult3 = BehaviorBuilder.get(p_260239_);
             return BehaviorBuilder.create(
-                new BehaviorBuilder.TriggerWithResult<E, R>(this) {
+                new BehaviorBuilder.TriggerWithResult<E, R>() {
                     @Override
                     public R tryTrigger(ServerLevel p_259096_, E p_260221_, long p_259035_) {
                         T1 t1 = triggerwithresult.tryTrigger(p_259096_, p_260221_, p_259035_);
@@ -291,7 +291,7 @@
             final BehaviorBuilder.TriggerWithResult<E, T4> triggerwithresult3 = BehaviorBuilder.get(p_259136_);
             final BehaviorBuilder.TriggerWithResult<E, Function4<T1, T2, T3, T4, R>> triggerwithresult4 = BehaviorBuilder.get(p_259519_);
             return BehaviorBuilder.create(
-                new BehaviorBuilder.TriggerWithResult<E, R>(this) {
+                new BehaviorBuilder.TriggerWithResult<E, R>() {
                     @Override
                     public R tryTrigger(ServerLevel p_259537_, E p_259581_, long p_259423_) {
                         T1 t1 = triggerwithresult.tryTrigger(p_259537_, p_259581_, p_259423_);
