--- a/net/minecraft/client/renderer/entity/EvokerRenderer.java
+++ b/net/minecraft/client/renderer/entity/EvokerRenderer.java
@@ -17,7 +17,7 @@
     public EvokerRenderer(EntityRendererProvider.Context p_174108_) {
         super(p_174108_, new IllagerModel<>(p_174108_.bakeLayer(ModelLayers.EVOKER)), 0.5F);
         this.addLayer(
-            new ItemInHandLayer<T, IllagerModel<T>>(this, this, p_174108_.getItemInHandRenderer()) {
+            new ItemInHandLayer<T, IllagerModel<T>>(this, p_174108_.getItemInHandRenderer()) {
                 public void render(
                     PoseStack p_114569_,
                     MultiBufferSource p_114570_,
