--- a/net/minecraft/world/item/enchantment/ConditionalEffect.java
+++ b/net/minecraft/world/item/enchantment/ConditionalEffect.java
@@ -20,7 +20,7 @@
                     ValidationContext validationcontext = new ValidationContext(problemreporter$collector, p_345335_);
                     p_351949_.validate(validationcontext);
                     return problemreporter$collector.getReport()
-                        .map(p_344978_ -> DataResult.error(() -> "Validation error in enchantment effect condition: " + p_344978_))
+                        .map(p_344978_ -> DataResult.<LootItemCondition>error(() -> "Validation error in enchantment effect condition: " + p_344978_))
                         .orElseGet(() -> DataResult.success(p_351949_));
                 }
             );
