--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -571,7 +_,7 @@
     }
 
     private static <S extends StateHolder<?, S>, T extends Comparable<T>> S setValueHelper(S state, Property<T> property, Object value) {
-        return state.setValue(property, (Comparable)value);
+        return state.setValue(property, (T) value);
     }
 
     @Deprecated
