--- a/net/minecraft/world/level/block/ChiseledBookShelfBlock.java
+++ b/net/minecraft/world/level/block/ChiseledBookShelfBlock.java
@@ -147,7 +_,7 @@
     @Override
     protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
         builder.add(FACING);
-        List var10000 = SLOT_OCCUPIED_PROPERTIES;
+        var var10000 = SLOT_OCCUPIED_PROPERTIES;
         Objects.requireNonNull(builder);
         StateDefinition.Builder var2 = builder;
         var10000.forEach(xva$0 -> var2.add(xva$0));
