--- a/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
@@ -109,19 +109,19 @@
             )
         );
         p_439608_.popPose();
-        f2 = -p_445495_;
+        float f2_f = -p_445495_;
         float f3 = -p_445495_;
-        f4 = -p_445495_;
-        f5 = -p_445495_;
+        float f4_f = -p_445495_;
+        float f5_f = -p_445495_;
         f9 = 0.0F;
         f10 = 1.0F;
-        f11 = -1.0F + f1;
-        f12 = p_440009_ * p_439343_ + f11;
+        float f11_f = -1.0F + f1;
+        float f12_f = p_440009_ * p_439343_ + f11;
         p_439393_.submitCustomGeometry(
             p_439608_,
             RenderType.beaconBeam(p_446739_, true),
             (p_440366_, p_440684_) -> renderPart(
-                p_440366_, p_440684_, ARGB.color(32, p_440282_), p_439067_, i, f2, f3, p_445495_, f4, f5, p_445495_, p_445495_, p_445495_, 0.0F, 1.0F, f12, f11
+                p_440366_, p_440684_, ARGB.color(32, p_440282_), p_439067_, i, f2_f, f3, p_445495_, f4_f, f5_f, p_445495_, p_445495_, p_445495_, 0.0F, 1.0F, f12_f, f11_f
             )
         );
         p_439608_.popPose();
