--- a/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
+++ b/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
@@ -10,7 +10,7 @@
 public class StopHoldingItemIfNoLongerAdmiring {
     public static BehaviorControl<Piglin> create() {
         return BehaviorBuilder.create(
-            p_259197_ -> p_259197_.<MemoryAccessor>group(p_259197_.absent(MemoryModuleType.ADMIRING_ITEM))
+            p_259197_ -> p_259197_.group(p_259197_.absent(MemoryModuleType.ADMIRING_ITEM))
                     .apply(p_259197_, p_259512_ -> (p_311695_, p_311696_, p_311697_) -> {
                             if (!p_311696_.getOffhandItem().isEmpty() && !p_311696_.getOffhandItem().is(Items.SHIELD)) {
                                 PiglinAi.stopHoldingOffHandItem(p_311696_, true);
