--- a/net/minecraft/client/gui/screens/telemetry/TelemetryInfoScreen.java
+++ b/net/minecraft/client/gui/screens/telemetry/TelemetryInfoScreen.java
@@ -70,8 +70,7 @@
         LinearLayout linearlayout4 = this.layout.addToContents(LinearLayout.vertical().spacing(8));
         this.telemetryEventWidget = linearlayout4.addChild(new TelemetryEventWidget(0, 0, this.width - 40, this.layout.getContentHeight(), this.font));
         this.telemetryEventWidget.setOnScrolledListener(p_262168_ -> this.savedScroll = p_262168_);
-        this.layout.visitWidgets(p_321366_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
