--- a/net/minecraft/data/loot/BlockLootSubProvider.java
+++ b/net/minecraft/data/loot/BlockLootSubProvider.java
@@ -345,7 +345,7 @@
          p_252139_,
          this.m_246108_(
             p_252139_,
-            ((LootPoolSingletonContainer.Builder)LootItem.m_79579_(Items.f_42404_).m_79080_(LootItemRandomChanceCondition.m_81927_(0.125F)))
+            LootItem.m_79579_(Items.f_42404_).m_79080_(LootItemRandomChanceCondition.m_81927_(0.125F))
                .m_79078_(ApplyBonusCount.m_79921_(Enchantments.f_44987_, 2))
          )
       );
@@ -397,7 +397,7 @@
                .m_79076_(
                   this.m_246108_(
                      p_249088_,
-                     ((LootPoolSingletonContainer.Builder)((LootPoolSingletonContainer.Builder)LootItem.m_79579_(p_249088_).m_79080_(p_251535_))
+                     LootItem.m_79579_(p_249088_).m_79080_(p_251535_)
                            .m_230987_(
                               Direction.values(),
                               p_251536_ -> SetItemCountFunction.m_165414_(ConstantValue.m_165692_(1.0F), true)
@@ -405,7 +405,7 @@
                                        LootItemBlockStatePropertyCondition.m_81769_(p_249088_)
                                           .m_81784_(StatePropertiesPredicate.Builder.m_67693_().m_67703_(MultifaceBlock.m_153933_(p_251536_), true))
                                     )
-                           ))
+                           )
                         .m_79078_(SetItemCountFunction.m_165414_(ConstantValue.m_165692_(-1.0F), true))
                   )
                )
