--- a/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
@@ -15,7 +15,7 @@
 public class AssignProfessionFromJobSite {
     public static BehaviorControl<Villager> create() {
         return BehaviorBuilder.create(
-            p_258312_ -> p_258312_.<MemoryAccessor, MemoryAccessor>group(
+            p_258312_ -> p_258312_.group(
                         p_258312_.present(MemoryModuleType.POTENTIAL_JOB_SITE), p_258312_.registered(MemoryModuleType.JOB_SITE)
                     )
                     .apply(
