--- a/net/minecraft/world/entity/ai/behavior/BackUpIfTooClose.java
+++ b/net/minecraft/world/entity/ai/behavior/BackUpIfTooClose.java
@@ -12,7 +12,7 @@
 public class BackUpIfTooClose {
     public static OneShot<Mob> create(int p_259782_, float p_259071_) {
         return BehaviorBuilder.create(
-            p_260278_ -> p_260278_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_260278_ -> p_260278_.group(
                         p_260278_.absent(MemoryModuleType.WALK_TARGET),
                         p_260278_.registered(MemoryModuleType.LOOK_TARGET),
                         p_260278_.present(MemoryModuleType.ATTACK_TARGET),
