--- a/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
+++ b/net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
@@ -13,7 +13,7 @@
         Predicate<E> p_260270_, MemoryModuleType<? extends T> p_260344_, MemoryModuleType<T> p_260014_, UniformInt p_259596_
     ) {
         return BehaviorBuilder.create(
-            p_260141_ -> p_260141_.<MemoryAccessor, MemoryAccessor>group(p_260141_.present(p_260344_), p_260141_.absent(p_260014_))
+            p_260141_ -> p_260141_.group(p_260141_.present(p_260344_), p_260141_.absent(p_260014_))
                     .apply(p_260141_, (p_259306_, p_259907_) -> (p_264887_, p_264888_, p_264889_) -> {
                             if (!p_260270_.test(p_264888_)) {
                                 return false;
