--- a/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
+++ b/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
@@ -15,7 +15,7 @@
 public class MeleeAttack {
     public static OneShot<Mob> create(int p_259758_) {
         return BehaviorBuilder.create(
-            p_258526_ -> p_258526_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258526_ -> p_258526_.group(
                         p_258526_.registered(MemoryModuleType.LOOK_TARGET),
                         p_258526_.present(MemoryModuleType.ATTACK_TARGET),
                         p_258526_.absent(MemoryModuleType.ATTACK_COOLING_DOWN),
