--- a/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
+++ b/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
@@ -16,7 +16,7 @@
 public class PoiCompetitorScan {
     public static BehaviorControl<Villager> create() {
         return BehaviorBuilder.create(
-            p_258576_ -> p_258576_.<MemoryAccessor, MemoryAccessor>group(
+            p_258576_ -> p_258576_.group(
                         p_258576_.present(MemoryModuleType.JOB_SITE), p_258576_.present(MemoryModuleType.NEAREST_LIVING_ENTITIES)
                     )
                     .apply(
