--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -17,7 +17,7 @@
 
     public static BehaviorControl<LivingEntity> create() {
         return BehaviorBuilder.create(
-            p_259094_ -> p_259094_.<MemoryAccessor>group(p_259094_.present(MemoryModuleType.MEETING_POINT))
+            p_259094_ -> p_259094_.group(p_259094_.present(MemoryModuleType.MEETING_POINT))
                     .apply(p_259094_, p_259028_ -> (p_259026_, p_260317_, p_260205_) -> {
                             if (p_259026_.random.nextFloat() <= 0.95F) {
                                 return false;
