--- a/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
+++ b/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
@@ -15,7 +15,7 @@
 
     public static OneShot<LivingEntity> create() {
         return BehaviorBuilder.create(
-            p_258755_ -> p_258755_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258755_ -> p_258755_.group(
                         p_258755_.registered(MemoryModuleType.WALK_TARGET),
                         p_258755_.registered(MemoryModuleType.LOOK_TARGET),
                         p_258755_.present(MemoryModuleType.MEETING_POINT),
