--- a/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
+++ b/net/minecraft/world/entity/ai/behavior/StartCelebratingIfTargetDead.java
@@ -12,7 +12,7 @@
 public class StartCelebratingIfTargetDead {
     public static BehaviorControl<LivingEntity> create(int p_259238_, BiPredicate<LivingEntity, LivingEntity> p_259102_) {
         return BehaviorBuilder.create(
-            p_259600_ -> p_259600_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_259600_ -> p_259600_.group(
                         p_259600_.present(MemoryModuleType.ATTACK_TARGET),
                         p_259600_.registered(MemoryModuleType.ANGRY_AT),
                         p_259600_.absent(MemoryModuleType.CELEBRATE_LOCATION),
