--- a/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
+++ b/net/minecraft/world/entity/ai/behavior/TryLaySpawnOnWaterNearLand.java
@@ -17,7 +17,7 @@
 public class TryLaySpawnOnWaterNearLand {
     public static BehaviorControl<LivingEntity> create(Block p_259207_) {
         return BehaviorBuilder.create(
-            p_259781_ -> p_259781_.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_259781_ -> p_259781_.group(
                         p_259781_.absent(MemoryModuleType.ATTACK_TARGET),
                         p_259781_.present(MemoryModuleType.WALK_TARGET),
                         p_259781_.present(MemoryModuleType.IS_PREGNANT)
