--- a/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -23,7 +23,7 @@
 
     public static OneShot<PathfinderMob> create(float p_259320_, int p_259708_, int p_259311_) {
         return BehaviorBuilder.create(
-            p_258869_ -> p_258869_.<MemoryAccessor>group(p_258869_.absent(MemoryModuleType.WALK_TARGET))
+            p_258869_ -> p_258869_.group(p_258869_.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         p_258869_,
                         p_258863_ -> (p_258874_, p_258875_, p_258876_) -> {
