--- a/net/minecraft/world/level/Explosion.java
+++ b/net/minecraft/world/level/Explosion.java
@@ -16,10 +16,10 @@
     @Nullable
     static LivingEntity getIndirectSourceEntity(@Nullable Entity p_363745_) {
         return switch (p_363745_) {
-            case null, default -> null;
             case PrimedTnt primedtnt -> primedtnt.getOwner();
             case LivingEntity livingentity -> livingentity;
             case Projectile projectile when projectile.getOwner() instanceof LivingEntity livingentity1 -> livingentity1;
+            case null, default -> null;
         };
     }
 
