--- a/net/minecraft/util/EncoderCache.java
+++ b/net/minecraft/util/EncoderCache.java
@@ -35,7 +_,7 @@
             public <T> DataResult<T> encode(A input, DynamicOps<T> ops, T prefix) {
                 return EncoderCache.this.cache
                     .getUnchecked(new EncoderCache.Key<>(codec, input, ops))
-                    .map(value -> value instanceof Tag tag ? tag.copy() : value);
+                    .map(value -> (T)(value instanceof Tag tag ? tag.copy() : value));
             }
         };
     }
