--- a/net/minecraft/util/Util.java
+++ b/net/minecraft/util/Util.java
@@ -531,7 +_,7 @@
     }
 
     public static <K, V1, V2> Map<K, V2> mapValues(final Map<K, V1> map, final Function<? super V1, V2> valueMapper) {
-        return map.entrySet().stream().collect((Collector) Collectors.toMap(Map.Entry::getKey, e -> valueMapper.apply(e.getValue())));
+        return map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> valueMapper.apply(e.getValue())));
     }
 
     public static <K, V1, V2> Map<K, V2> mapValuesLazy(final Map<K, V1> map, final com.google.common.base.Function<V1, V2> valueMapper) {
