--- a/net/minecraft/world/entity/AgeableMob.java
+++ b/net/minecraft/world/entity/AgeableMob.java
@@ -71,12 +71,13 @@
 
     public void ageUp(int p_146741_, boolean p_146742_) {
         int i = this.getAge();
+        int k = i;
         i += p_146741_ * 20;
         if (i > 0) {
             i = 0;
         }
 
-        int j = i - i;
+        int j = i - k;
         this.setAge(i);
         if (p_146742_) {
             this.forcedAge += j;
