--- a/net/minecraft/client/resources/model/BlockStateModelLoader.java
+++ b/net/minecraft/client/resources/model/BlockStateModelLoader.java
@@ -214,7 +214,7 @@
                 Property<?> property = p_352099_.getProperty(s1);
                 if (property != null && iterator.hasNext()) {
                     String s2 = iterator.next();
-                    Comparable<?> comparable = getValueHelper((Property<Comparable<?>>)property, s2);
+                    Comparable<?> comparable = getValueHelper(property, s2);
                     if (comparable == null) {
                         throw new RuntimeException("Unknown value: '" + s2 + "' for blockstate property: '" + s1 + "' " + property.getPossibleValues());
                     }
