--- a/net/minecraft/world/entity/Leashable.java
+++ b/net/minecraft/world/entity/Leashable.java
@@ -43,7 +43,7 @@
 
     default void setDelayedLeashHolderId(int p_352387_) {
         this.setLeashData(new Leashable.LeashData(p_352387_));
-        dropLeash((Entity)this, false, false);
+        dropLeash((Entity & Leashable)this, false, false);
     }
 
     default void readLeashData(CompoundTag p_390528_) {
@@ -82,11 +82,11 @@
     }
 
     default void dropLeash() {
-        dropLeash((Entity)this, true, true);
+        dropLeash((Entity & Leashable)this, true, true);
     }
 
     default void removeLeash() {
-        dropLeash((Entity)this, true, false);
+        dropLeash((Entity & Leashable)this, true, false);
     }
 
     default void onLeashRemoved() {
@@ -155,7 +155,7 @@
     }
 
     default void elasticRangeLeashBehaviour(Entity p_353036_, float p_353047_) {
-        legacyElasticRangeLeashBehaviour((Entity)this, p_353036_, p_353047_);
+        legacyElasticRangeLeashBehaviour((Entity & Leashable)this, p_353036_, p_353047_);
     }
 
     private static <E extends Entity & Leashable> void legacyElasticRangeLeashBehaviour(E p_353048_, Entity p_353039_, float p_353053_) {
@@ -168,7 +168,7 @@
     }
 
     default void setLeashedTo(Entity p_352411_, boolean p_352183_) {
-        setLeashedTo((Entity)this, p_352411_, p_352183_);
+        setLeashedTo((Entity & Leashable)this, p_352411_, p_352183_);
     }
 
     private static <E extends Entity & Leashable> void setLeashedTo(E p_352280_, Entity p_352109_, boolean p_352239_) {
@@ -191,7 +191,7 @@
 
     @Nullable
     default Entity getLeashHolder() {
-        return getLeashHolder((Entity)this);
+        return getLeashHolder((Entity & Leashable)this);
     }
 
     @Nullable
