--- a/net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
@@ -293,7 +_,7 @@
             private final boolean canReport;
             private final boolean playerMessage;
 
-            public MessageEntry(int chatId, Component text, @Nullable Component narration, GuiMessageTag tag, boolean canReport, boolean playerMessage) {
+            public MessageEntry(int chatId, Component text, Component narration, @Nullable GuiMessageTag tag, boolean canReport, boolean playerMessage) {
                 this.chatId = chatId;
                 this.tagIcon = Optionull.map(tag, GuiMessageTag::icon);
                 this.tagHoverText = tag != null && tag.text() != null
