--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -245,7 +245,7 @@
                 P p1 = p_184445_.get(1);
                 return p_184365_.apply(p, p1);
             }), p_184459_ -> ImmutableList.of(p_184366_.apply(p_184459_), p_184367_.apply(p_184459_)));
-        Codec<I> codec1 = RecordCodecBuilder.<Pair>create(
+        Codec<I> codec1 = RecordCodecBuilder.<Pair<P, P>>create(
                 p_184360_ -> p_184360_.group(p_184362_.fieldOf(p_184363_).forGetter(Pair::getFirst), p_184362_.fieldOf(p_184364_).forGetter(Pair::getSecond))
                         .apply(p_184360_, Pair::of)
             )
@@ -624,8 +624,8 @@
                 return p_342443_;
             }
 
-            private <T, V2 extends V> DataResult<T> encode(Codec<V2> p_331770_, V p_331859_, DynamicOps<T> p_331717_) {
-                return p_331770_.encodeStart(p_331717_, p_331859_);
+            private <T, V2 extends V> DataResult<T> encode(Codec<? extends V> p_331770_, V p_331859_, DynamicOps<T> p_331717_) {
+                return ((Codec<V2>)p_331770_).encodeStart(p_331717_, (V2)p_331859_);
             }
         };
     }
@@ -645,7 +645,7 @@
             }
 
             private <T, V2 extends V> DataResult<T> encodeValue(Codec<V2> p_342234_, V p_341931_, DynamicOps<T> p_341769_) {
-                return p_342234_.encodeStart(p_341769_, p_341931_);
+                return p_342234_.encodeStart(p_341769_, (V2)p_341931_);
             }
 
             @Override
