--- a/com/mojang/realmsclient/gui/screens/RealmsParentalConsentScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsParentalConsentScreen.java
@@ -35,8 +35,7 @@
         Component component = Component.translatable("mco.account.privacy.info.button");
         linearlayout.addChild(Button.builder(component, ConfirmLinkScreen.confirmLink(this, "https://aka.ms/MinecraftGDPR")).build());
         linearlayout.addChild(Button.builder(CommonComponents.GUI_BACK, p_307025_ -> this.onClose()).build());
-        this.layout.visitWidgets(p_307027_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
