--- a/net/minecraft/client/gui/screens/worldselection/EditWorldScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/EditWorldScreen.java
@@ -113,8 +113,7 @@
             );
         this.layout.addChild(new SpacerElement(200, 20));
         this.layout.addChild(linearlayout);
-        this.layout.visitWidgets(p_321379_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
     }
 
     @Override
