--- a/net/minecraft/commands/ParserUtils.java
+++ b/net/minecraft/commands/ParserUtils.java
@@ -47,7 +47,7 @@
         Object object;
         try {
             JsonElement jsonelement = Streams.parse(jsonreader);
-            object = p_313876_.parse(p_323581_.createSerializationContext(JsonOps.INSTANCE), (T)jsonelement).getOrThrow(JsonParseException::new);
+            object = p_313876_.parse(p_323581_.createSerializationContext(JsonOps.INSTANCE), jsonelement).getOrThrow(JsonParseException::new);
         } catch (StackOverflowError stackoverflowerror) {
             throw new JsonParseException(stackoverflowerror);
         } finally {
