--- a/net/minecraft/client/main/Main.java
+++ b/net/minecraft/client/main/Main.java
@@ -194,7 +194,7 @@
         };
         thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(LOGGER));
         Runtime.getRuntime().addShutdownHook(thread);
-        final Minecraft minecraft = null;
+        Minecraft minecraft = null;
 
         try {
             Thread.currentThread().setName("Render thread");
@@ -223,7 +223,7 @@
                 public void run() {
                     try {
                         RenderSystem.initGameThread(true);
-                        minecraft.run();
+                        minecraft1.run();
                     } catch (Throwable throwable3) {
                         Main.LOGGER.error("Exception in client thread", throwable3);
                     }
