--- a/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
@@ -121,8 +121,7 @@
             }
         }
 
-        this.layout.visitWidgets(p_321350_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
@@ -287,8 +286,8 @@
                                     } else if (this.cancelled) {
                                         this.uploadCancelled();
                                     } else {
-                                        i = (int)(new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
-                                        file1 = this.tarGzipArchive(new File(i, this.selectedLevel.getLevelId()));
+                                        File file = (new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
+                                        file1 = this.tarGzipArchive(new File(file, this.selectedLevel.getLevelId()));
                                         if (this.cancelled) {
                                             this.uploadCancelled();
                                         } else if (this.verify(file1)) {
@@ -357,13 +356,10 @@
                                 }
                             }
                         } catch (IOException ioexception) {
-                            uploadinfo = ioexception;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", ioexception.getMessage()));
                         } catch (RealmsServiceException realmsserviceexception) {
-                            uploadinfo = realmsserviceexception;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", realmsserviceexception.realmsError.errorMessage()));
                         } catch (InterruptedException interruptedexception1) {
-                            uploadinfo = interruptedexception1;
                             LOGGER.error("Could not acquire upload lock");
                         }
                     } finally {
