--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -393,7 +393,7 @@
     }
 
     public void openScreen(ChatComponent.ChatMethod p_437427_, ChatScreen.ChatConstructor<?> p_437274_) {
-        this.minecraft.setScreen(this.createScreen(p_437427_, (ChatScreen.ChatConstructor<Screen>)p_437274_));
+        this.minecraft.setScreen(this.createScreen(p_437427_, p_437274_));
     }
 
     public void preserveCurrentChatScreen() {
@@ -463,7 +463,7 @@
         COMMAND("/") {
             @Override
             public boolean isDraftRestorable(ChatComponent.Draft p_437306_) {
-                return this == p_437306_.chatMethod;
+                return this == p_437306_.chatMethod();
             }
         };
 
