--- a/net/minecraft/client/telemetry/TelemetryPropertyMap.java
+++ b/net/minecraft/client/telemetry/TelemetryPropertyMap.java
@@ -61,7 +61,7 @@
                T t = p_261668_.get(p_261627_.f_260687_());
                if (t != null) {
                   DataResult<V> dataresult = p_261627_.f_260706_().parse(p_261859_, t);
-                  return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.m_261137_(p_261627_, (T)p_261796_), dataresult);
+                  return p_261892_.apply2stable((p_262028_, p_261796_) -> p_262028_.m_261137_((TelemetryProperty<T>)p_261627_, (T)p_261796_), dataresult);
                } else {
                   return p_261892_;
                }
