--- a/net/minecraft/client/gui/screens/worldselection/WorldOpenFlows.java
+++ b/net/minecraft/client/gui/screens/worldselection/WorldOpenFlows.java
@@ -140,7 +140,10 @@
    public Pair<LevelSettings, WorldCreationContext> m_246225_(LevelStorageSource.LevelStorageAccess p_249540_) throws Exception {
       PackRepository packrepository = ServerPacksSource.m_247728_(p_249540_);
       WorldLoader.PackConfig worldloader$packconfig = this.m_247188_(p_249540_, false, packrepository);
-      return this.m_246486_(
+      @OnlyIn(Dist.CLIENT)
+      record Data(LevelSettings f_244166_, WorldOptions f_244534_, Registry<LevelStem> f_244151_) {
+      }
+      return this.<Data, Pair<LevelSettings, WorldCreationContext>>m_246486_(
          worldloader$packconfig,
          p_247857_ -> {
             DynamicOps<Tag> dynamicops = RegistryOps.m_255058_(NbtOps.f_128958_, p_247857_.f_244104_());
@@ -165,12 +168,8 @@
          }
       );
 
-      @OnlyIn(Dist.CLIENT)
-      record Data(LevelSettings f_244166_, WorldOptions f_244534_, Registry<LevelStem> f_244151_) {
       }
 
-   }
-
    private WorldLoader.PackConfig m_247188_(LevelStorageSource.LevelStorageAccess p_249986_, boolean p_248615_, PackRepository p_249167_) {
       WorldDataConfiguration worlddataconfiguration = p_249986_.m_247706_();
       if (worlddataconfiguration == null) {
