--- a/net/minecraft/util/valueproviders/UniformInt.java
+++ b/net/minecraft/util/valueproviders/UniformInt.java
@@ -9,7 +_,7 @@
 
 public class UniformInt extends IntProvider {
 
-    public static final MapCodec<UniformInt> CODEC = RecordCodecBuilder.mapCodec(i -> i.group(Codec.INT.fieldOf("min_inclusive").forGetter(u -> Integer.valueOf(u.minInclusive)), Codec.INT.fieldOf("max_inclusive").forGetter(u -> Integer.valueOf(u.maxInclusive))).apply(i, UniformInt::new)).validate(u -> {
+    public static final MapCodec<UniformInt> CODEC = RecordCodecBuilder.<UniformInt>mapCodec(i -> i.group(Codec.INT.fieldOf("min_inclusive").forGetter(u -> Integer.valueOf(u.minInclusive)), Codec.INT.fieldOf("max_inclusive").forGetter(u -> Integer.valueOf(u.maxInclusive))).apply(i, UniformInt::new)).validate(u -> {
         if (u.maxInclusive < u.minInclusive) {
             return DataResult.error(() -> "Max must be at least min, min_inclusive: " + u.minInclusive + ", max_inclusive: " + u.maxInclusive);
         }
