--- a/net/minecraft/core/RegistryAccess.java
+++ b/net/minecraft/core/RegistryAccess.java
@@ -34,8 +34,8 @@
       return new RegistryAccess.Frozen() {
          @Override
          public <T> Optional<Registry<T>> m_6632_(ResourceKey<? extends Registry<? extends T>> p_206220_) {
-            Registry<Registry<T>> registry = p_206166_;
-            return registry.m_123009_(p_206220_);
+            Registry<Registry<T>> registry = (Registry<Registry<T>>)p_206166_;
+            return registry.m_123009_((ResourceKey<Registry<T>>)p_206220_);
          }
 
          @Override
@@ -72,7 +72,7 @@
 
       public ImmutableRegistryAccess(List<? extends Registry<?>> p_248540_) {
          this.f_206223_ = p_248540_.stream()
-            .collect(Collectors.toUnmodifiableMap(Registry::m_123023_, (Function<? super Registry, ? extends Registry>)(p_206232_ -> p_206232_)));
+            .collect(Collectors.toUnmodifiableMap(Registry::m_123023_, (p_206232_ -> p_206232_)));
       }
 
       public ImmutableRegistryAccess(Map<? extends ResourceKey<? extends Registry<?>>, ? extends Registry<?>> p_206225_) {
@@ -85,7 +85,7 @@
 
       @Override
       public <E> Optional<Registry<E>> m_6632_(ResourceKey<? extends Registry<? extends E>> p_206229_) {
-         return Optional.ofNullable(this.f_206223_.get(p_206229_)).map((Function<? super Registry<?>, ? extends Registry<E>>)(p_247993_ -> p_247993_));
+         return Optional.ofNullable(this.f_206223_.get(p_206229_)).map((Function<? super Registry<?>, ? extends Registry<E>>)(p_247993_ -> (Registry<E>)p_247993_));
       }
 
       @Override
@@ -102,7 +102,7 @@
       }
 
       private static <T> RegistryAccess.RegistryEntry<T> m_206243_(ResourceKey<? extends Registry<?>> p_206244_, Registry<?> p_206245_) {
-         return new RegistryAccess.RegistryEntry<>(p_206244_, p_206245_);
+         return new RegistryAccess.RegistryEntry<T>((ResourceKey<? extends Registry<T>>)p_206244_, (Registry<T>)p_206245_);
       }
 
       private RegistryAccess.RegistryEntry<T> m_206247_() {
