--- a/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
@@ -15,7 +15,7 @@
 public class GoToClosestVillage {
    public static BehaviorControl<Villager> m_257375_(float p_260342_, int p_259691_) {
       return BehaviorBuilder.m_258034_(
-         p_258357_ -> p_258357_.<MemoryAccessor>group(p_258357_.m_258080_(MemoryModuleType.f_26370_))
+         p_258357_ -> p_258357_.group(p_258357_.m_258080_(MemoryModuleType.f_26370_))
                .apply(p_258357_, p_258366_ -> (p_274970_, p_274971_, p_274972_) -> {
                      if (p_274970_.m_8802_(p_274971_.m_20183_())) {
                         return false;
