--- a/net/minecraft/world/phys/shapes/EntityCollisionContext.java
+++ b/net/minecraft/world/phys/shapes/EntityCollisionContext.java
@@ -47,7 +47,7 @@
             p_198920_ instanceof LivingEntity livingentity ? livingentity.getMainHandItem() : ItemStack.EMPTY,
             p_198916_
                 ? p_360273_ -> true
-                : (p_198920_ instanceof LivingEntity livingentity1 ? p_404624_ -> livingentity1.canStandOnFluid(p_404624_) : p_205113_ -> false),
+                : (p_198920_ instanceof LivingEntity livingentity1 ? p_404624_ -> ((LivingEntity) p_198920_).canStandOnFluid(p_404624_) : p_205113_ -> false),
             p_198920_
         );
     }
