--- a/net/minecraft/world/level/storage/loot/functions/CopyComponentsFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/CopyComponentsFunction.java
@@ -125,18 +125,19 @@
         }
 
         public DataComponentMap get(LootContext p_331014_) {
-            switch (this) {
-                case BLOCK_ENTITY:
+            return switch (this) {
+                case BLOCK_ENTITY -> {
                     BlockEntity blockentity = p_331014_.getParamOrNull(LootContextParams.BLOCK_ENTITY);
-                    return blockentity != null ? blockentity.collectComponents() : DataComponentMap.EMPTY;
-            }
+                    yield blockentity != null ? blockentity.collectComponents() : DataComponentMap.EMPTY;
+                }
+            };
         }
 
         public Set<LootContextParam<?>> getReferencedContextParams() {
-            switch (this) {
-                case BLOCK_ENTITY:
-                    return Set.of(LootContextParams.BLOCK_ENTITY);
-            }
+            return switch (this) {
+                case BLOCK_ENTITY ->
+                    Set.of(LootContextParams.BLOCK_ENTITY);
+            };
         }
 
         @Override
