--- a/net/minecraft/world/item/crafting/SelectableRecipe.java
+++ b/net/minecraft/world/item/crafting/SelectableRecipe.java
@@ -17,7 +17,7 @@
 
     public record SingleInputEntry<T extends Recipe<?>>(Ingredient input, SelectableRecipe<T> recipe) {
         public static <T extends Recipe<?>> StreamCodec<RegistryFriendlyByteBuf, SelectableRecipe.SingleInputEntry<T>> noRecipeCodec() {
-            return StreamCodec.composite(
+            return StreamCodec.<RegistryFriendlyByteBuf, SingleInputEntry<T>, Ingredient, SelectableRecipe<T>>composite(
                 Ingredient.CONTENTS_STREAM_CODEC,
                 SelectableRecipe.SingleInputEntry::input,
                 SelectableRecipe.noRecipeCodec(),
