--- a/com/mojang/realmsclient/gui/task/DataFetcher.java
+++ b/com/mojang/realmsclient/gui/task/DataFetcher.java
@@ -52,7 +52,7 @@
         private final Consumer<T> output;
         private long lastCheckTime = -1L;
 
-        SubscribedTask(DataFetcher.Task<T> var1, Consumer<T> p_239959_, Consumer p_239960_) {
+        SubscribedTask(DataFetcher.Task<T> p_239959_, Consumer<T> p_239960_) {
             this.task = p_239959_;
             this.output = p_239960_;
         }
@@ -89,7 +89,7 @@
         private final List<DataFetcher.SubscribedTask<?>> subscriptions = new ArrayList<>();
 
         public <T> void subscribe(DataFetcher.Task<T> p_239442_, Consumer<T> p_239443_) {
-            DataFetcher.SubscribedTask<T> subscribedtask = DataFetcher.this.new SubscribedTask<>(DataFetcher.this, p_239442_, p_239443_);
+            DataFetcher.SubscribedTask<T> subscribedtask = DataFetcher.this.new SubscribedTask<>(p_239442_, p_239443_);
             this.subscriptions.add(subscribedtask);
             subscribedtask.runCallbackIfNeeded();
         }
