--- a/net/minecraft/util/datafix/DataFixers.java
+++ b/net/minecraft/util/datafix/DataFixers.java
@@ -1016,7 +1016,7 @@
         Schema schema161 = p_14514_.addSchema(3086, SAME_NAMESPACED);
         p_14514_.addFixer(
             new EntityVariantFix(
-                schema161, "Change cat variant type", References.ENTITY, "minecraft:cat", "CatType", Util.make(new Int2ObjectOpenHashMap(), p_216528_ -> {
+                schema161, "Change cat variant type", References.ENTITY, "minecraft:cat", "CatType", Util.make(new Int2ObjectOpenHashMap<String>(), p_216528_ -> {
                     p_216528_.defaultReturnValue("minecraft:tabby");
                     p_216528_.put(0, "minecraft:tabby");
                     p_216528_.put(1, "minecraft:black");
@@ -1056,7 +1056,7 @@
         Schema schema162 = p_14514_.addSchema(3087, SAME_NAMESPACED);
         p_14514_.addFixer(
             new EntityVariantFix(
-                schema162, "Change frog variant type", References.ENTITY, "minecraft:frog", "Variant", Util.make(new Int2ObjectOpenHashMap(), p_216519_ -> {
+                schema162, "Change frog variant type", References.ENTITY, "minecraft:frog", "Variant", Util.make(new Int2ObjectOpenHashMap<String>(), p_216519_ -> {
                     p_216519_.put(0, "minecraft:temperate");
                     p_216519_.put(1, "minecraft:warm");
                     p_216519_.put(2, "minecraft:cold");
