--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -234,7 +234,7 @@
         default Function<OptionInstance<T>, AbstractWidget> createButton(
             OptionInstance.TooltipSupplier<T> p_261801_, Options p_261824_, int p_261649_, int p_262114_, int p_261536_, Consumer<T> p_261642_
         ) {
-            return p_454137_ -> CycleButton.builder(p_454137_.toString, p_454137_::get)
+            return p_454137_ -> CycleButton.<T>builder(p_454137_.toString, p_454137_::get)
                 .withValues(this.valueListSupplier())
                 .withTooltip(p_261801_)
                 .create(p_261649_, p_262114_, p_261536_, 20, p_454137_.caption, (p_261347_, p_261348_) -> {
