--- a/net/minecraft/world/entity/monster/warden/Warden.java
+++ b/net/minecraft/world/entity/monster/warden/Warden.java
@@ -563,7 +563,7 @@
             protected PathFinder createPathFinder(int p_219479_) {
                 this.nodeEvaluator = new WalkNodeEvaluator();
                 this.nodeEvaluator.setCanPassDoors(true);
-                return new PathFinder(this, this.nodeEvaluator, p_219479_) {
+                return new PathFinder(this.nodeEvaluator, p_219479_) {
                     @Override
                     protected float distance(Node p_219486_, Node p_219487_) {
                         return p_219486_.distanceToXZ(p_219487_);
