--- a/net/minecraft/world/entity/monster/piglin/StopAdmiringIfTiredOfTryingToReachItem.java
+++ b/net/minecraft/world/entity/monster/piglin/StopAdmiringIfTiredOfTryingToReachItem.java
@@ -11,7 +11,7 @@
 public class StopAdmiringIfTiredOfTryingToReachItem {
     public static BehaviorControl<LivingEntity> create(int p_259110_, int p_259200_) {
         return BehaviorBuilder.create(
-            p_260320_ -> p_260320_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_260320_ -> p_260320_.group(
                         p_260320_.present(MemoryModuleType.ADMIRING_ITEM),
                         p_260320_.present(MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM),
                         p_260320_.registered(MemoryModuleType.TIME_TRYING_TO_REACH_ADMIRE_ITEM),
