--- a/net/minecraft/world/entity/pets/PetAxolotl.java
+++ b/net/minecraft/world/entity/pets/PetAxolotl.java
@@ -145,7 +145,7 @@
             SpawnGroupData petaxolotl$axolotlgroupdata = new PetAxolotl.AxolotlGroupData(
                 Axolotl.Variant.getCommonSpawnVariant(randomsource), Axolotl.Variant.getCommonSpawnVariant(randomsource)
             );
-            this.setVariant(petaxolotl$axolotlgroupdata.getVariant(randomsource));
+            this.setVariant(((PetAxolotl.AxolotlGroupData)petaxolotl$axolotlgroupdata).getVariant(randomsource));
             return super.finalizeSpawn(p_421677_, p_421531_, p_422859_, petaxolotl$axolotlgroupdata);
         }
     }
