--- a/net/minecraft/client/gui/narration/ScreenNarrationCollector.java
+++ b/net/minecraft/client/gui/narration/ScreenNarrationCollector.java
@@ -21,7 +21,7 @@
 
     public String collectNarrationText(boolean p_169189_) {
         final StringBuilder stringbuilder = new StringBuilder();
-        Consumer<String> consumer = new Consumer<String>(this) {
+        Consumer<String> consumer = new Consumer<String>() {
             private boolean firstEntry = true;
 
             public void accept(String p_169204_) {
