--- a/net/minecraft/world/level/levelgen/SurfaceRules.java
+++ b/net/minecraft/world/level/levelgen/SurfaceRules.java
@@ -753,14 +753,14 @@
 
                 @Override
                 protected boolean compute() {
-                    int i = this.context.blockY;
-                    if (i <= i) {
+                    int k = this.context.blockY;
+                    if (k <= i) {
                         return true;
-                    } else if (i >= j) {
+                    } else if (k >= j) {
                         return false;
                     } else {
-                        double d0 = Mth.map((double)i, (double)i, (double)j, 1.0, 0.0);
-                        RandomSource randomsource = positionalrandomfactory.at(this.context.blockX, i, this.context.blockZ);
+                        double d0 = Mth.map((double)k, (double)i, (double)j, 1.0, 0.0);
+                        RandomSource randomsource = positionalrandomfactory.at(this.context.blockX, k, this.context.blockZ);
                         return (double)randomsource.nextFloat() < d0;
                     }
                 }
