--- a/net/minecraft/world/inventory/RecipeBookMenu.java
+++ b/net/minecraft/world/inventory/RecipeBookMenu.java
@@ -13,7 +13,7 @@
     }
 
     public void handlePlacement(boolean p_40119_, RecipeHolder<?> p_300860_, ServerPlayer p_40121_) {
-        new ServerPlaceRecipe<>(this).recipeClicked(p_40121_, p_300860_, p_40119_);
+        new ServerPlaceRecipe<>(this).recipeClicked(p_40121_, (RecipeHolder<? extends Recipe<C>>)p_300860_, p_40119_);
     }
 
     public abstract void fillCraftSlotsStackedContents(StackedContents p_40117_);
