--- a/net/minecraft/data/tags/ItemTagsProvider.java
+++ b/net/minecraft/data/tags/ItemTagsProvider.java
@@ -42,7 +42,7 @@
         return super.createContentsProvider().thenCombine(this.blockTags, (p_274766_, p_274767_) -> {
             this.tagsToCopy.forEach((p_274763_, p_274764_) -> {
                 TagBuilder tagbuilder = this.getOrCreateRawBuilder((TagKey<Item>)p_274764_);
-                Optional<TagBuilder> optional = p_274767_.apply((TagKey<? super TagKey<Block>>)p_274763_);
+                Optional<TagBuilder> optional = p_274767_.apply(p_274763_);
                 optional.orElseThrow(() -> new IllegalStateException("Missing block tag " + p_274764_.location())).build().forEach(tagbuilder::add);
             });
             return (HolderLookup.Provider)p_274766_;
