--- a/net/minecraft/world/entity/ai/behavior/ReactToBell.java
+++ b/net/minecraft/world/entity/ai/behavior/ReactToBell.java
@@ -11,7 +11,7 @@
 public class ReactToBell {
    public static BehaviorControl<LivingEntity> m_258068_() {
       return BehaviorBuilder.m_258034_(
-         p_259349_ -> p_259349_.<MemoryAccessor>group(p_259349_.m_257495_(MemoryModuleType.f_26325_))
+         p_259349_ -> p_259349_.group(p_259349_.m_257495_(MemoryModuleType.f_26325_))
                .apply(p_259349_, p_259472_ -> (p_293915_, p_293916_, p_293917_) -> {
                      Raid raid = p_293915_.m_8832_(p_293916_.m_20183_());
                      if (raid == null) {
