--- a/net/minecraft/client/gui/screens/Screen.java
+++ b/net/minecraft/client/gui/screens/Screen.java
@@ -260,18 +260,7 @@
                 Objects.requireNonNull(clickevent);
                 Throwable throwable5;
                 switch (clickevent) {
-                    case ClickEvent.OpenUrl clickevent$openurl:
-                        ClickEvent.OpenUrl clickevent$openurl1 = clickevent$openurl;
-
-                        try {
-                            uri1 = clickevent$openurl1.uri();
-                        } catch (Throwable throwable4) {
-                            throwable5 = throwable4;
-                            boolean flag2 = false;
-                            break;
-                        }
-
-                        URI uri = uri1;
+                    case ClickEvent.OpenUrl(URI uri):
                         if (!this.minecraft.options.chatLinks().get()) {
                             return false;
                         }
@@ -292,32 +281,10 @@
                     case ClickEvent.OpenFile clickevent$openfile:
                         Util.getPlatform().openFile(clickevent$openfile.file());
                         return true;
-                    case ClickEvent.SuggestCommand clickevent$suggestcommand:
-                        ClickEvent.SuggestCommand clickevent$suggestcommand1 = clickevent$suggestcommand;
-
-                        try {
-                            s6 = clickevent$suggestcommand1.command();
-                        } catch (Throwable throwable3) {
-                            throwable5 = throwable3;
-                            boolean flag1 = false;
-                            break;
-                        }
-
-                        String s2 = s6;
+                    case ClickEvent.SuggestCommand(String s2):
                         this.insertText(s2, true);
                         return true;
-                    case ClickEvent.RunCommand clickevent$runcommand:
-                        ClickEvent.RunCommand clickevent$runcommand1 = clickevent$runcommand;
-
-                        try {
-                            s5 = clickevent$runcommand1.command();
-                        } catch (Throwable throwable2) {
-                            throwable5 = throwable2;
-                            boolean flag = false;
-                            break;
-                        }
-
-                        String s3 = s5;
+                    case ClickEvent.RunCommand(String s3):
                         String s = s3;
                         if (s3.startsWith("/")) {
                             s = s3.substring(1);
@@ -328,27 +295,13 @@
                         }
 
                         return true;
-                    case ClickEvent.CopyToClipboard clickevent$copytoclipboard:
-                        ClickEvent.CopyToClipboard clickevent$copytoclipboard1 = clickevent$copytoclipboard;
-
-                        try {
-                            s4 = clickevent$copytoclipboard1.value();
-                        } catch (Throwable throwable1) {
-                            throwable5 = throwable1;
-                            boolean flag3 = false;
-                            break;
-                        }
-
-                        String s1 = s4;
+                    case ClickEvent.CopyToClipboard(String s1):
                         this.minecraft.keyboardHandler.setClipboard(s1);
                         return true;
                     default:
                         LOGGER.error("Don't know how to handle {}", clickevent);
                         return true;
                 }
-
-                Throwable throwable = throwable5;
-                throw new MatchException(throwable.toString(), throwable);
             }
 
             return false;
