--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -101,7 +101,7 @@
         if (p_114383_ instanceof AbstractClientPlayer abstractclientplayer) {
             PlayerSkin.Model playerskin$model = abstractclientplayer.getSkin().model();
             EntityRenderer<? extends Player> entityrenderer = this.playerRenderers.get(playerskin$model);
-            return entityrenderer != null ? entityrenderer : this.playerRenderers.get(PlayerSkin.Model.WIDE);
+            return (EntityRenderer<? super T>)(entityrenderer != null ? entityrenderer : this.playerRenderers.get(PlayerSkin.Model.WIDE));
         } else {
             return (EntityRenderer<? super T>)this.renderers.get(p_114383_.getType());
         }
