--- a/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
@@ -22,7 +22,7 @@
         return p_14890_.get("Owner")
             .get()
             .map(p_14894_ -> replaceUUIDString(p_14894_, "Id", "Id").orElse(p_14894_))
-            .map(p_14888_ -> p_14890_.remove("Owner").set("SkullOwner", p_14888_))
+            .<Dynamic<?>>map(p_14888_ -> p_14890_.remove("Owner").set("SkullOwner", p_14888_))
             .result()
             .orElse(p_14890_);
     }
