--- a/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundUpdateRecipesPacket.java
@@ -47,6 +47,6 @@
     public static <T extends Recipe<?>> void toNetwork(FriendlyByteBuf p_179470_, RecipeHolder<?> p_301050_) {
         p_179470_.writeResourceLocation(BuiltInRegistries.RECIPE_SERIALIZER.getKey(p_301050_.value().getSerializer()));
         p_179470_.writeResourceLocation(p_301050_.id());
-        p_301050_.value().getSerializer().toNetwork(p_179470_, (T)p_301050_.value());
+        ((net.minecraft.world.item.crafting.RecipeSerializer<T>)p_301050_.value().getSerializer()).toNetwork(p_179470_, (T)p_301050_.value());
     }
 }
