--- a/net/minecraft/world/attribute/modifier/FloatModifier.java
+++ b/net/minecraft/world/attribute/modifier/FloatModifier.java
@@ -23,11 +23,11 @@
             );
         }
     };
-    FloatModifier<Float> ADD = Float::sum;
+    FloatModifier<Float> ADD = (Simple) Float::sum;
     FloatModifier<Float> SUBTRACT = (FloatModifier.Simple)(p_458036_, p_458119_) -> p_458036_ - p_458119_;
     FloatModifier<Float> MULTIPLY = (FloatModifier.Simple)(p_457586_, p_458256_) -> p_457586_ * p_458256_;
-    FloatModifier<Float> MINIMUM = Math::min;
-    FloatModifier<Float> MAXIMUM = Math::max;
+    FloatModifier<Float> MINIMUM = (Simple) Math::min;
+    FloatModifier<Float> MAXIMUM = (Simple) Math::max;
 
     @FunctionalInterface
     public interface Simple extends FloatModifier<Float> {
