--- a/net/minecraft/util/datafix/fixes/BlockEntityRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityRenameFix.java
@@ -20,8 +20,8 @@
 
    @Override
    public TypeRewriteRule makeRule() {
-      TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.f_16781_);
-      TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.f_16781_);
+      TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>)this.getInputSchema().findChoiceType(References.f_16781_);
+      TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>)this.getOutputSchema().findChoiceType(References.f_16781_);
       return this.fixTypeEverywhere(this.f_276471_, taggedchoicetype, taggedchoicetype1, p_277946_ -> p_277512_ -> p_277512_.mapFirst(this.f_276461_));
    }
 
