--- a/net/minecraft/util/datafix/fixes/EntityIdFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityIdFix.java
@@ -97,8 +97,8 @@
 
    @Override
    public TypeRewriteRule makeRule() {
-      TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.f_16786_);
-      TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>)this.getInputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>)this.getOutputSchema().findChoiceType(References.f_16786_);
       Type<?> type = this.getInputSchema().getType(References.f_16782_);
       Type<?> type1 = this.getOutputSchema().getType(References.f_16782_);
       return TypeRewriteRule.seq(
