--- a/com/mojang/realmsclient/gui/screens/RealmsPlayerScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsPlayerScreen.java
@@ -60,8 +60,7 @@
                 .build()
         );
         linearlayout.addChild(Button.builder(CommonComponents.GUI_BACK, p_329657_ -> this.onClose()).build());
-        this.layout.visitWidgets(p_329656_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
