--- a/net/minecraft/data/Main.java
+++ b/net/minecraft/data/Main.java
@@ -200,13 +200,14 @@
             bindRegistries(RegistriesDatapackGenerator::new, completablefuture3.thenApply(RegistrySetBuilder.PatchedRegistries::patches))
         );
         datagenerator$packgenerator3.addProvider(bindRegistries(WinterDropRecipeProvider.Runner::new, completablefuture));
+        CompletableFuture<HolderLookup.Provider> completablefuture_f = completablefuture;
         TagsProvider<Block> tagsprovider4 = datagenerator$packgenerator3.addProvider(
-            p_378854_ -> new WinterDropBlockTagsProvider(p_378854_, completablefuture, tagsprovider5.contentsGetter())
+            p_378854_ -> new WinterDropBlockTagsProvider(p_378854_, completablefuture_f, tagsprovider5.contentsGetter())
         );
         datagenerator$packgenerator3.addProvider(
-            p_378850_ -> new WinterDropItemTagsProvider(p_378850_, completablefuture, tagsprovider.contentsGetter(), tagsprovider4.contentsGetter())
+            p_378850_ -> new WinterDropItemTagsProvider(p_378850_, completablefuture_f, tagsprovider.contentsGetter(), tagsprovider4.contentsGetter())
         );
-        datagenerator$packgenerator3.addProvider(p_378846_ -> new WinterDropBiomeTagsProvider(p_378846_, completablefuture, tagsprovider1.contentsGetter()));
+        datagenerator$packgenerator3.addProvider(p_378846_ -> new WinterDropBiomeTagsProvider(p_378846_, completablefuture_f, tagsprovider1.contentsGetter()));
         datagenerator$packgenerator3.addProvider(bindRegistries(WinterDropLootTableProvider::create, completablefuture));
         datagenerator$packgenerator3.addProvider(
             p_378851_ -> PackMetadataGenerator.forFeaturePack(
