--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -137,11 +137,11 @@
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(500, 600),
                 new StageMinusOne<>(6000, Pose.SLEEPING),
-                new StageZero(1200, new Vec3(-3.0, 0.0, 7.0), Pose.EMERGING),
-                new StageOne(1200, new Vec3(0.0, 0.0, -14.0), Pose.STANDING),
-                new StageTwo(1200, new Vec3(0.0, 0.0, 7.0), Pose.SLEEPING),
-                new StageThree(1200, new Vec3(3.0, 0.0, 0.0), Pose.ROARING),
-                new StageFour(60, Pose.DIGGING)
+                new StageZero<>(1200, new Vec3(-3.0, 0.0, 7.0), Pose.EMERGING),
+                new StageOne<>(1200, new Vec3(0.0, 0.0, -14.0), Pose.STANDING),
+                new StageTwo<>(1200, new Vec3(0.0, 0.0, 7.0), Pose.SLEEPING),
+                new StageThree<>(1200, new Vec3(3.0, 0.0, 0.0), Pose.ROARING),
+                new StageFour<>(60, Pose.DIGGING)
             ),
             ImmutableSet.of(
                 Pair.of(MemoryModuleType.ACTING_STAGE, MemoryStatus.VALUE_PRESENT), Pair.of(MemoryModuleType.ATTACK_TARGET, MemoryStatus.VALUE_ABSENT)
