--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -502,9 +502,10 @@
 
         int k = Math.max(0, this.unloadQueue.size() - 2000);
 
-        while((p_140354_.getAsBoolean() || k > 0) && (j = (long)this.unloadQueue.poll()) != null) {
+        Runnable runnable;
+        while((p_140354_.getAsBoolean() || k > 0) && (runnable = this.unloadQueue.poll()) != null) {
             --k;
-            j.run();
+            runnable.run();
         }
 
         int l = 0;
@@ -589,7 +590,7 @@
                 }
 
                 return flag;
-            })).thenApplyAsync(p_313584_ -> {
+            })).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync(p_313584_ -> {
             this.level.getProfiler().incrementCounter("chunkLoad");
             if (p_313584_.isPresent()) {
                 ChunkAccess chunkaccess = ChunkSerializer.read(this.level, this.poiManager, p_140418_, p_313584_.get());
