--- a/com/mojang/realmsclient/gui/screens/configuration/RealmsPreferredRegionSelectionScreen.java
+++ b/com/mojang/realmsclient/gui/screens/configuration/RealmsPreferredRegionSelectionScreen.java
@@ -120,7 +_,7 @@
             private final RealmsSettingsTab.RegionSelection regionSelection;
             private final Component name;
 
-            public Entry(@Nullable RegionSelectionPreference preference, RealmsRegion region) {
+            public Entry(RegionSelectionPreference preference, @Nullable RealmsRegion region) {
                 this(new RealmsSettingsTab.RegionSelection(preference, region));
             }
 
