--- a/net/minecraft/util/datafix/fixes/ItemShulkerBoxColorFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemShulkerBoxColorFix.java
@@ -58,7 +_,7 @@
                             Dynamic<?> blockEntityRest = blockEntity.get(DSL.remainderFinder());
                             int color = blockEntityRest.get("Color").asInt(0);
                             blockEntityRest.remove("Color");
-                            return input.set(tagF, tag.set(blockEntityF, blockEntity.set(DSL.remainderFinder(), blockEntityRest)))
+                            return input.set(tagF, (Typed<?>) tag.set(blockEntityF, blockEntity.set(DSL.remainderFinder(), blockEntityRest)))
                                 .set(idF, Pair.of(References.ITEM_NAME.typeName(), NAMES_BY_COLOR[color % 16]));
                         }
                     }
