--- a/net/minecraft/world/entity/ai/behavior/Mount.java
+++ b/net/minecraft/world/entity/ai/behavior/Mount.java
@@ -13,7 +13,7 @@
 
     public static BehaviorControl<LivingEntity> create(float p_259363_) {
         return BehaviorBuilder.create(
-            p_259880_ -> p_259880_.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_259880_ -> p_259880_.group(
                         p_259880_.registered(MemoryModuleType.LOOK_TARGET),
                         p_259880_.absent(MemoryModuleType.WALK_TARGET),
                         p_259880_.present(MemoryModuleType.RIDE_TARGET)
