--- a/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -56,7 +56,7 @@
         String p_309198_, FixProjectileStoredItem.SubFixer<?> p_309081_, Type<?> p_309063_, Type<T> p_308929_
     ) {
         OpticFinder<?> opticfinder = DSL.namedChoice(p_309198_, p_309063_);
-        return p_309195_ -> p_309195_.updateTyped(opticfinder, p_308929_, p_309191_ -> p_309081_.fix(p_309191_, p_308929_));
+        return p_309195_ -> p_309195_.updateTyped(opticfinder, p_308929_, p_309191_ -> ((FixProjectileStoredItem.SubFixer<T>)p_309081_).fix(p_309191_, p_308929_));
     }
 
     private static <T> Typed<T> fixArrow(Typed<?> p_308973_, Type<T> p_309096_) {
