--- a/net/minecraft/world/phys/shapes/CollisionContext.java
+++ b/net/minecraft/world/phys/shapes/CollisionContext.java
@@ -38,7 +38,7 @@
             true,
             p_405836_ != null ? p_405836_.getY() : -Double.MAX_VALUE,
             p_405836_ instanceof LivingEntity livingentity ? livingentity.getMainHandItem() : ItemStack.EMPTY,
-            p_405836_ instanceof LivingEntity livingentity1 ? p_404827_ -> livingentity1.canStandOnFluid(p_404827_) : p_405865_ -> false,
+            p_405836_ instanceof LivingEntity livingentity1 ? p_404827_ -> ((LivingEntity) p_405836_).canStandOnFluid(p_404827_) : p_405865_ -> false,
             p_405836_
         );
     }
