--- a/net/minecraft/server/jsonrpc/api/Schema.java
+++ b/net/minecraft/server/jsonrpc/api/Schema.java
@@ -40,15 +_,15 @@
             "Schema",
             subCodec -> RecordCodecBuilder.create(
                 i -> i.group(
-                        (App<Mu<? extends Schema<?>>, Optional<URI>>)ReferenceUtil.REFERENCE_CODEC.optionalFieldOf("$ref").forGetter(Schema::reference),
-                        (App<Mu<? extends Schema<?>>, List<String>>)ExtraCodecs.compactListCodec(Codec.STRING)
+                        ReferenceUtil.REFERENCE_CODEC.optionalFieldOf("$ref").forGetter(Schema::reference),
+                        ExtraCodecs.compactListCodec(Codec.STRING)
                             .optionalFieldOf("type", List.of())
                             .forGetter(Schema::type),
-                        (App<Mu<? extends Schema<?>>, Optional<? extends Schema<?>>>)subCodec.optionalFieldOf("items").forGetter(Schema::items),
-                        (App<Mu<? extends Schema<?>>, Map<String, ? extends Schema<?>>>)Codec.unboundedMap(Codec.STRING, subCodec)
+                        subCodec.optionalFieldOf("items").forGetter(Schema::items),
+                        Codec.unboundedMap(Codec.STRING, subCodec)
                             .optionalFieldOf("properties", Map.of())
                             .forGetter(Schema::properties),
-                        (App<Mu<? extends Schema<?>>, List<String>>)Codec.STRING.listOf().optionalFieldOf("enum", List.of()).forGetter(Schema::enumValues)
+                        Codec.STRING.listOf().optionalFieldOf("enum", List.of()).forGetter(Schema::enumValues)
                     )
                     .apply(i, (ref, type, items, properties, enumValues) -> null)
             )
