--- a/net/minecraft/world/entity/ai/gossip/GossipContainer.java
+++ b/net/minecraft/world/entity/ai/gossip/GossipContainer.java
@@ -162,7 +162,7 @@
             .decode(p_26178_)
             .resultOrPartial(p_262901_ -> LOGGER.warn("Failed to deserialize gossips: {}", p_262901_))
             .stream()
-            .flatMap(p_262899_ -> ((List)p_262899_.getFirst()).stream())
+            .flatMap(p_262899_ -> (p_262899_.getFirst()).stream())
             .forEach(p_26162_ -> this.getOrCreate(p_26162_.target).entries.put(p_26162_.type, p_26162_.value));
     }
 
