--- a/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java
@@ -103,7 +103,7 @@
     }
 
     public Component getTestName() {
-        return this.test().map(p_397722_ -> Component.literal(p_397722_.location().toString())).orElse(INVALID_TEST_NAME);
+        return this.test().map(p_397722_ -> (Component)Component.literal(p_397722_.location().toString())).orElse(INVALID_TEST_NAME);
     }
 
     private Optional<Holder.Reference<GameTestInstance>> getTestHolder() {
