--- a/net/minecraft/server/commands/DebugCommand.java
+++ b/net/minecraft/server/commands/DebugCommand.java
@@ -127,7 +127,7 @@
                         try {
                             CommandSourceStack commandsourcestack = p_309716_.withSource(debugcommand$tracer).withMaximumPermission(2);
                             InstantiatedFunction<CommandSourceStack> instantiatedfunction = commandfunction.instantiate(null, commanddispatcher);
-                            p_306283_.queueNext((new CallFunction<CommandSourceStack>(this, instantiatedfunction, CommandResultCallback.EMPTY, false) {
+                            p_306283_.queueNext((new CallFunction<CommandSourceStack>(instantiatedfunction, CommandResultCallback.EMPTY, false) {
                                 public void execute(CommandSourceStack p_309660_, ExecutionContext<CommandSourceStack> p_309654_, Frame p_309674_) {
                                     printwriter.println(commandfunction.id());
                                     super.execute(p_309660_, p_309654_, p_309674_);
