--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
@@ -15,7 +15,7 @@
 public class SetWalkTargetFromBlockMemory {
     public static OneShot<Villager> create(MemoryModuleType<GlobalPos> p_259685_, float p_259842_, int p_259530_, int p_260360_, int p_259504_) {
         return BehaviorBuilder.create(
-            p_258717_ -> p_258717_.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258717_ -> p_258717_.group(
                         p_258717_.registered(MemoryModuleType.CANT_REACH_WALK_TARGET_SINCE),
                         p_258717_.absent(MemoryModuleType.WALK_TARGET),
                         p_258717_.present(p_259685_)
