--- a/net/minecraft/util/datafix/fixes/MobSpawnerEntityIdentifiersFix.java
+++ b/net/minecraft/util/datafix/fixes/MobSpawnerEntityIdentifiersFix.java
@@ -64,7 +64,7 @@
             Dynamic<?> dynamic = p_337649_.get(DSL.remainderFinder());
             dynamic = dynamic.set("id", dynamic.createString("MobSpawner"));
             DataResult<? extends Pair<? extends Typed<?>, ?>> dataresult = type.readTyped(this.fix(dynamic));
-            return dataresult.result().isEmpty() ? p_337649_ : ((Pair)dataresult.result().get()).getFirst();
+            return dataresult.result().isEmpty() ? p_337649_ : (dataresult.result().get()).getFirst();
         });
     }
 }
