--- a/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
@@ -24,15 +24,15 @@
             p_16648_ -> p_16648_.update(
                     DSL.remainderFinder(),
                     p_337655_ -> p_337655_.getMapValues().map(p_145588_ -> p_337655_.createMap(p_145588_.entrySet().stream().map(p_145585_ -> {
-                                if (((Dynamic)p_145585_.getKey()).asString("").startsWith("key_")) {
-                                    String s = ((Dynamic)p_145585_.getValue()).asString("");
+                                if ((p_145585_.getKey()).asString("").startsWith("key_")) {
+                                    String s = (p_145585_.getValue()).asString("");
                                     if (!s.startsWith("key.mouse") && !s.startsWith("scancode.")) {
-                                        return Pair.of((Dynamic)p_145585_.getKey(), p_337655_.createString("key.keyboard." + s.substring("key.".length())));
+                                        return Pair.of(p_145585_.getKey(), p_337655_.createString("key.keyboard." + s.substring("key.".length())));
                                     }
                                 }
         
-                                return Pair.of((Dynamic)p_145585_.getKey(), (Dynamic)p_145585_.getValue());
-                            }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse(p_337655_)
+                                return Pair.of(p_145585_.getKey(), (Dynamic)p_145585_.getValue());
+                            }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)))).result().orElse((Dynamic)p_337655_)
                 )
         );
     }
