--- a/net/minecraft/client/multiplayer/ClientLevel.java
+++ b/net/minecraft/client/multiplayer/ClientLevel.java
@@ -1134,6 +1134,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.addAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
@@ -1147,6 +1148,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.removeAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
