--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -440,7 +_,7 @@
     }
 
     public void openScreen(ChatComponent.ChatMethod chatMethod, ChatScreen.ChatConstructor<?> chat) {
-        this.minecraft.setScreen(this.createScreen(chatMethod, (ChatScreen.ChatConstructor<Screen>)chat));
+        this.minecraft.setScreen(this.createScreen(chatMethod, chat));
     }
 
     public void preserveCurrentChatScreen() {
