--- a/net/minecraft/client/gui/screens/telemetry/TelemetryInfoScreen.java
+++ b/net/minecraft/client/gui/screens/telemetry/TelemetryInfoScreen.java
@@ -78,8 +78,7 @@
         linearlayout.addChild(layout);
         framelayout.arrangeElements();
         FrameLayout.alignInRectangle(framelayout, 0, 0, this.width, this.height, 0.5F, 0.0F);
-        framelayout.visitWidgets(p_264696_ -> {
-        });
+        framelayout.visitWidgets(this::addRenderableWidget);
     }
 
     private Layout createLowerSection() {
