--- a/net/minecraft/client/particle/SuspendedParticle.java
+++ b/net/minecraft/client/particle/SuspendedParticle.java
@@ -86,7 +86,7 @@
             double p_172436_,
             double p_172437_
         ) {
-            SuspendedParticle suspendedparticle = new SuspendedParticle(this, p_172431_, this.sprite, p_172432_, p_172433_, p_172434_, 0.0, -0.8F, 0.0) {
+            SuspendedParticle suspendedparticle = new SuspendedParticle(p_172431_, this.sprite, p_172432_, p_172433_, p_172434_, 0.0, -0.8F, 0.0) {
                 @Override
                 public Optional<ParticleGroup> getParticleGroup() {
                     return Optional.of(ParticleGroup.SPORE_BLOSSOM);
