--- a/net/minecraft/world/entity/ai/behavior/BabyFollowAdult.java
+++ b/net/minecraft/world/entity/ai/behavior/BabyFollowAdult.java
@@ -17,7 +17,7 @@
 
     public static OneShot<AgeableMob> create(UniformInt p_259321_, Function<LivingEntity, Float> p_259190_) {
         return BehaviorBuilder.create(
-            p_258331_ -> p_258331_.<MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258331_ -> p_258331_.group(
                         p_258331_.present(MemoryModuleType.NEAREST_VISIBLE_ADULT),
                         p_258331_.registered(MemoryModuleType.LOOK_TARGET),
                         p_258331_.absent(MemoryModuleType.WALK_TARGET)
