--- a/net/minecraft/world/entity/ai/behavior/InteractWith.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWith.java
@@ -30,7 +30,7 @@
       int i = p_259564_ * p_259564_;
       Predicate<LivingEntity> predicate = p_293871_ -> p_259366_.equals(p_293871_.m_6095_()) && p_260254_.test((T)p_293871_);
       return BehaviorBuilder.m_258034_(
-         p_258426_ -> p_258426_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+         p_258426_ -> p_258426_.group(
                   p_258426_.m_257492_(p_260229_),
                   p_258426_.m_257492_(MemoryModuleType.f_26371_),
                   p_258426_.m_258080_(MemoryModuleType.f_26370_),
@@ -45,7 +45,7 @@
                               p_258419_ -> p_258419_.m_20280_(p_258414_) <= (double)i && predicate.test(p_258419_)
                            );
                            optional.ifPresent(p_258432_ -> {
-                              p_258439_.m_257512_(p_258432_);
+                              p_258439_.m_257512_((T)p_258432_);
                               p_258440_.m_257512_(new EntityTracker(p_258432_, true));
                               p_258441_.m_257512_(new WalkTarget(new EntityTracker(p_258432_, false), p_259369_, p_259065_));
                            });
