--- a/net/minecraft/client/input/InputWithModifiers.java
+++ b/net/minecraft/client/input/InputWithModifiers.java
@@ -87,6 +_,7 @@
 
     @Retention(RetentionPolicy.CLASS)
     @Target(ElementType.TYPE_USE)
+    @org.intellij.lang.annotations.MagicConstant(flags = {InputConstants.MOD_SHIFT, InputConstants.MOD_CONTROL, InputConstants.MOD_ALT, InputConstants.MOD_SUPER, InputConstants.MOD_CAPS_LOCK, InputConstants.MOD_NUM_LOCK})
     @interface Modifiers {
     }
 }
