--- a/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
@@ -40,8 +_,8 @@
         return TypeRewriteRule.seq(
             this.fixTypeEverywhere(
                 "InjectBedBlockEntityType",
-                this.getInputSchema().findChoiceType(References.BLOCK_ENTITY),
-                this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
+                (Type) this.getInputSchema().findChoiceType(References.BLOCK_ENTITY),
+                (Type) this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
                 ops -> v -> v
             ),
             this.fixTypeEverywhereTyped(
