--- a/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -46,7 +_,7 @@
         String entityName, FixProjectileStoredItem.SubFixer<?> fixer, Type<?> inputEntityChoiceType, Type<T> outputEntityChoiceType
     ) {
         OpticFinder<?> entityF = DSL.namedChoice(entityName, inputEntityChoiceType);
-        return input -> input.updateTyped(entityF, outputEntityChoiceType, typed -> fixer.fix(typed, outputEntityChoiceType));
+        return input -> input.updateTyped(entityF, outputEntityChoiceType, typed -> ((FixProjectileStoredItem.SubFixer<T>)fixer).fix(typed, outputEntityChoiceType));
     }
 
     private static <T> Typed<T> fixArrow(Typed<?> typed, Type<T> outputType) {
