--- a/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
+++ b/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
@@ -21,14 +21,14 @@
                         p_259791_,
                         (p_259255_, p_260214_, p_259562_, p_259156_) -> (p_375937_, p_375938_, p_375939_) -> {
                                 if (!p_375938_.isBaby()
-                                    && !p_259791_.<List>tryGet(p_259156_)
+                                    && !p_259791_.tryGet(p_259156_)
                                         .map(p_259958_ -> p_259958_.stream().anyMatch(StartHuntingHoglin::hasHuntedRecently))
                                         .isPresent()) {
                                     Hoglin hoglin = p_259791_.get(p_259255_);
                                     PiglinAi.setAngerTarget(p_375937_, p_375938_, hoglin);
                                     PiglinAi.dontKillAnyMoreHoglinsForAWhile(p_375938_);
                                     PiglinAi.broadcastAngerTarget(p_375937_, p_375938_, hoglin);
-                                    p_259791_.<List>tryGet(p_259156_).ifPresent(p_259760_ -> p_259760_.forEach(PiglinAi::dontKillAnyMoreHoglinsForAWhile));
+                                    p_259791_.tryGet(p_259156_).ifPresent(p_259760_ -> p_259760_.forEach(PiglinAi::dontKillAnyMoreHoglinsForAWhile));
                                     return true;
                                 } else {
                                     return false;
