--- a/net/minecraft/client/network/webrtc/RtcChannel.java
+++ b/net/minecraft/client/network/webrtc/RtcChannel.java
@@ -16,7 +_,6 @@
 import io.netty.channel.DefaultChannelConfig;
 import io.netty.channel.EventLoop;
 import io.netty.channel.SingleThreadEventLoop;
-import io.netty.channel.AbstractChannel.AbstractUnsafe;
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 import java.nio.ByteBuffer;
@@ -257,7 +_,6 @@
 
     private final class RtcUnsafe extends AbstractUnsafe {
         private RtcUnsafe() {
-            super(RtcChannel.this);
         }
 
         @Override
