--- a/net/minecraft/world/entity/ai/behavior/BecomePassiveIfMemoryPresent.java
+++ b/net/minecraft/world/entity/ai/behavior/BecomePassiveIfMemoryPresent.java
@@ -18,7 +18,7 @@
                         p_259944_,
                         p_259944_.point(
                             () -> "[BecomePassive if " + p_259988_ + " present]",
-                            (Function3<MemoryAccessor, MemoryAccessor, MemoryAccessor, Trigger<LivingEntity>>)(p_260120_, p_259674_, p_259822_) -> (p_260328_, p_259412_, p_259725_) -> {
+                            (p_260120_, p_259674_, p_259822_) -> (p_260328_, p_259412_, p_259725_) -> {
                                     p_259674_.setWithExpiry(true, (long)p_260143_);
                                     p_260120_.erase();
                                     return true;
