--- a/com/mojang/blaze3d/opengl/GlProgram.java
+++ b/com/mojang/blaze3d/opengl/GlProgram.java
@@ -79,7 +79,7 @@
         for (RenderPipeline.UniformDescription uniformDescription : uniforms) {
             String uniformName = uniformDescription.name();
 
-            Object var10000_1 = switch (uniformDescription.type()) {
+            Object var10000 = switch (uniformDescription.type()) {
                 case UNIFORM_BUFFER -> {
                     int index = GL31.glGetUniformBlockIndex(this.programId, uniformName);
                     if (index == -1) {
