--- a/net/minecraft/server/level/ChunkTaskPriorityQueue.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueue.java
@@ -94,7 +94,7 @@
                 this.firstQueue++;
             }
 
-            return list.stream().map(p_140529_ -> p_140529_.map(Either::left).orElseGet(() -> Either.right(this.acquire(j))));
+            return list.stream().map(p_140529_ -> p_140529_.<Either<T,Runnable>>map(Either::left).orElseGet(() -> Either.right(this.acquire(j))));
         }
     }
 
