--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -86,7 +86,7 @@
 
     @Nullable
     public static <StoredEntityType extends UniquelyIdentifyable> EntityReference<StoredEntityType> read(CompoundTag p_393801_, String p_394012_) {
-        return p_393801_.read(p_394012_, codec()).orElse(null);
+        return p_393801_.read(p_394012_, EntityReference.<StoredEntityType>codec()).orElse(null);
     }
 
     @Nullable
@@ -98,7 +98,7 @@
             ? new EntityReference<>(optional.get())
             : p_393884_.getString(p_394543_)
                 .map(p_409192_ -> OldUsersConverter.convertMobOwnerIfNecessary(p_393661_.getServer(), p_409192_))
-                .map(EntityReference::new)
+                .map(EntityReference<StoredEntityType>::new)
                 .orElse(null);
     }
 }
