--- a/net/minecraft/util/ArrayListDeque.java
+++ b/net/minecraft/util/ArrayListDeque.java
@@ -324,6 +324,10 @@
         return new ArrayListDeque.DescendingIterator();
     }
 
+    public ArrayListDeque<T> reversed() {
+        throw new UnsupportedOperationException("Method was added in java 21 and is required to be overridden for classes implementing both List and Deque");
+    }
+
     class DescendingIterator implements Iterator<T> {
         private int index = ArrayListDeque.this.size() - 1;
 
