--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -4080,6 +_,7 @@
 
     @Retention(RetentionPolicy.CLASS)
     @Target(ElementType.TYPE_USE)
+    @org.intellij.lang.annotations.MagicConstant(intValues = {FLAG_ONFIRE, FLAG_SHIFT_KEY_DOWN, FLAG_SPRINTING, FLAG_SWIMMING, FLAG_INVISIBLE, FLAG_GLOWING, FLAG_FALL_FLYING})
     public @interface Flags {
     }
 
