--- a/net/minecraft/util/datafix/fixes/BannerEntityCustomNameToOverrideComponentFix.java
+++ b/net/minecraft/util/datafix/fixes/BannerEntityCustomNameToOverrideComponentFix.java
@@ -25,7 +25,7 @@
         TaggedChoiceType<?> taggedchoicetype = this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
         OpticFinder<?> opticfinder = type.findField("components");
         return this.fixTypeEverywhereTyped("Banner entity custom_name to item_name component fix", type, p_338407_ -> {
-            Object object = p_338407_.<Pair>get(taggedchoicetype.finder()).getFirst();
+            Object object = p_338407_.get(taggedchoicetype.finder()).getFirst();
             return object.equals("minecraft:banner") ? this.fix(p_338407_, opticfinder) : p_338407_;
         });
     }
