--- a/net/minecraft/client/gui/screens/reporting/NameReportScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/NameReportScreen.java
@@ -59,8 +59,7 @@
       linearlayout.addChild(Button.builder(CommonComponents.GUI_BACK, p_301660_ -> this.onClose()).width(120).build());
       this.sendButton = linearlayout.addChild(Button.builder(SEND_REPORT, p_299852_ -> this.sendReport()).width(120).build());
       this.onReportChanged();
-      this.layout.visitWidgets(p_300038_ -> {
-      });
+      this.layout.visitWidgets(this::addRenderableWidget);
       this.repositionElements();
    }
 
