--- a/net/minecraft/client/gui/Gui.java
+++ b/net/minecraft/client/gui/Gui.java
@@ -567,7 +567,8 @@
 
                     p_421175_.blitSprite(RenderType::guiTextured, EFFECT_BACKGROUND_SPRITE, j1, k1, 24, 24);
                     int i2 = k1;
-                    list1.add(() -> p_421175_.renderFakeItem(WorldEffects.createEffectItem(worldeffect, false), j1 + 4, i2 + 4));
+                    final int j1_f = j1;
+                    list1.add(() -> p_421175_.renderFakeItem(WorldEffects.createEffectItem(worldeffect, false), j1_f + 4, i2 + 4));
                     if (screen != null && i >= j1 && j >= k1 && i < j1 + 24 && j < k1 + 24) {
                         screen.setTooltipForNextRenderPass(
                             CommonComponents.joinLines(worldeffect.name(), worldeffect.description().copy().withStyle(ChatFormatting.GRAY))
