--- a/net/minecraft/world/entity/ai/behavior/RandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/RandomStroll.java
@@ -45,7 +45,7 @@
 
     private static OneShot<PathfinderMob> strollFlyOrSwim(float p_260030_, Function<PathfinderMob, Vec3> p_259912_, Predicate<PathfinderMob> p_259088_) {
         return BehaviorBuilder.create(
-            p_258620_ -> p_258620_.<MemoryAccessor>group(p_258620_.absent(MemoryModuleType.WALK_TARGET))
+            p_258620_ -> p_258620_.group(p_258620_.absent(MemoryModuleType.WALK_TARGET))
                     .apply(p_258620_, p_258600_ -> (p_258610_, p_258611_, p_258612_) -> {
                             if (!p_259088_.test(p_258611_)) {
                                 return false;
