--- a/net/minecraft/world/level/levelgen/feature/TreeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/TreeFeature.java
@@ -134,7 +134,7 @@
             set1.add(p_160548_.immutable());
             worldgenlevel.setBlock(p_160548_, p_160549_, 19);
         };
-        FoliagePlacer.FoliageSetter foliageplacer$foliagesetter = new FoliagePlacer.FoliageSetter(this) {
+        FoliagePlacer.FoliageSetter foliageplacer$foliagesetter = new FoliagePlacer.FoliageSetter() {
             @Override
             public void set(BlockPos p_272825_, BlockState p_273311_) {
                 set2.add(p_272825_.immutable());
