--- a/net/minecraft/client/renderer/item/properties/select/ComponentContents.java
+++ b/net/minecraft/client/renderer/item/properties/select/ComponentContents.java
@@ -23,7 +23,7 @@
         Codec<? extends DataComponentType<?>> rawComponentCodec = BuiltInRegistries.DATA_COMPONENT_TYPE
             .byNameCodec()
             .validate(t -> t.isTransient() ? DataResult.error(() -> "Component can't be serialized") : DataResult.success(t));
-        MapCodec<SelectItemModel.UnbakedSwitch<ComponentContents<T>, T>> switchCodec = rawComponentCodec.dispatchMap(
+        MapCodec<SelectItemModel.UnbakedSwitch<ComponentContents<T>, T>> switchCodec = ((Codec<DataComponentType<T>>)rawComponentCodec).dispatchMap(
             "component",
             switchObject -> switchObject.property().componentType,
             componentType -> SelectItemModelProperty.Type.createCasesFieldCodec(componentType.codecOrThrow())
