--- a/net/minecraft/client/renderer/item/properties/select/ComponentContents.java
+++ b/net/minecraft/client/renderer/item/properties/select/ComponentContents.java
@@ -23,7 +23,7 @@
         Codec<? extends DataComponentType<?>> codec = BuiltInRegistries.DATA_COMPONENT_TYPE
             .byNameCodec()
             .validate(p_398049_ -> p_398049_.isTransient() ? DataResult.error(() -> "Component can't be serialized") : DataResult.success(p_398049_));
-        MapCodec<SelectItemModel.UnbakedSwitch<ComponentContents<T>, T>> mapcodec = codec.dispatchMap(
+        MapCodec<SelectItemModel.UnbakedSwitch<ComponentContents<T>, T>> mapcodec = ((Codec<DataComponentType<T>>)codec).dispatchMap(
             "component",
             p_397332_ -> p_397332_.property().componentType,
             p_397367_ -> SelectItemModelProperty.Type.createCasesFieldCodec(p_397367_.codecOrThrow())
