--- a/net/minecraft/client/gui/Font.java
+++ b/net/minecraft/client/gui/Font.java
@@ -144,7 +144,7 @@
     ) {
         int i = adjustColor(p_168650_);
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
+            p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
         );
 
         for (int j = -1; j <= 1; j++) {
@@ -168,7 +168,7 @@
 
         font$stringrenderoutput.renderCharacters();
         Font.StringRenderOutput font$stringrenderoutput1 = new Font.StringRenderOutput(
-            this, p_168652_, p_168647_, p_168648_, adjustColor(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
+            p_168652_, p_168647_, p_168648_, adjustColor(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
         );
         p_168646_.accept(font$stringrenderoutput1);
         font$stringrenderoutput1.finish(p_168647_);
@@ -240,7 +240,7 @@
         boolean p_381721_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_, p_381721_
+            p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_, p_381721_
         );
         StringDecomposer.iterateFormatted(p_381719_, Style.EMPTY, font$stringrenderoutput);
         return font$stringrenderoutput.finish(p_272632_);
@@ -260,7 +260,7 @@
         boolean p_381718_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_, p_381718_
+            p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_, p_381718_
         );
         p_381723_.accept(font$stringrenderoutput);
         return font$stringrenderoutput.finish(p_273532_);
@@ -343,7 +343,6 @@
         }
 
         public StringRenderOutput(
-            Font param1,
             MultiBufferSource p_181365_,
             float p_181366_,
             float p_181367_,
@@ -353,11 +352,10 @@
             Font.DisplayMode p_181371_,
             int p_181372_
         ) {
-            this(var1, p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_, true);
+            this(p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_, true);
         }
 
         public StringRenderOutput(
-            Font param1,
             MultiBufferSource p_379354_,
             float p_380215_,
             float p_380204_,
@@ -369,7 +367,6 @@
             int p_379625_,
             boolean p_381720_
         ) {
-            this.this$0 = var1;
             this.glyphInstances = new ArrayList<>();
             this.bufferSource = p_379354_;
             this.x = p_380215_;
@@ -386,8 +383,8 @@
 
         @Override
         public boolean accept(int p_92967_, Style p_92968_, int p_92969_) {
-            FontSet fontset = this.this$0.getFontSet(p_92968_.getFont());
-            GlyphInfo glyphinfo = fontset.getGlyphInfo(p_92969_, this.this$0.filterFishyGlyphs);
+            FontSet fontset = Font.this.getFontSet(p_92968_.getFont());
+            GlyphInfo glyphinfo = fontset.getGlyphInfo(p_92969_, Font.this.filterFishyGlyphs);
             BakedGlyph bakedglyph = p_92968_.isObfuscated() && p_92969_ != 32 ? fontset.getRandomGlyph(glyphinfo) : fontset.getGlyph(p_92969_);
             boolean flag = p_92968_.isBold();
             TextColor textcolor = p_92968_.getColor();
@@ -419,7 +416,7 @@
                 BakedGlyph.Effect bakedglyph$effect = new BakedGlyph.Effect(
                     p_381032_ - 1.0F, this.y + 9.0F, this.x, this.y - 1.0F, this.getUnderTextEffectDepth(), this.backgroundColor
                 );
-                bakedglyph = this.this$0.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
+                bakedglyph = Font.this.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
                 VertexConsumer vertexconsumer = this.bufferSource.getBuffer(bakedglyph.renderType(this.mode));
                 bakedglyph.renderEffect(bakedglyph$effect, this.pose, vertexconsumer, this.packedLightCoords);
             }
@@ -427,7 +424,7 @@
             this.renderCharacters();
             if (this.effects != null) {
                 if (bakedglyph == null) {
-                    bakedglyph = this.this$0.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
+                    bakedglyph = Font.this.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
                 }
 
                 VertexConsumer vertexconsumer1 = this.bufferSource.getBuffer(bakedglyph.renderType(this.mode));
