--- a/net/minecraft/world/entity/animal/Sheep.java
+++ b/net/minecraft/world/entity/animal/Sheep.java
@@ -86,10 +86,10 @@
         p_29841_.put(DyeColor.RED, Blocks.RED_WOOL);
         p_29841_.put(DyeColor.BLACK, Blocks.BLACK_WOOL);
     });
-    private static final Map<DyeColor, float[]> COLORARRAY_BY_COLOR = Maps.newEnumMap(
+    private static final Map<DyeColor, float[]> COLORARRAY_BY_COLOR = Maps.<DyeColor, float[]>newEnumMap(
         Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_29868_ -> p_29868_, Sheep::createSheepColor))
     );
-    private static final Map<DyeColor, float[]> COLORARRAY_BY_POTATO = Maps.newEnumMap(
+    private static final Map<DyeColor, float[]> COLORARRAY_BY_POTATO = Maps.<DyeColor, float[]>newEnumMap(
         Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_340118_ -> p_340118_, Sheep::createSheepColorPotato))
     );
     private int eatAnimationTick;
