--- a/net/minecraft/client/multiplayer/ClientLevel.java
+++ b/net/minecraft/client/multiplayer/ClientLevel.java
@@ -1097,6 +1097,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.addAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
@@ -1110,6 +1111,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.dragonParts.removeAll(Arrays.asList(enderdragon.getSubEntities()));
                     break;
+                default:
             }
         }
 
