--- a/net/minecraft/world/level/chunk/PalettedContainer.java
+++ b/net/minecraft/world/level/chunk/PalettedContainer.java
@@ -49,7 +_,7 @@
     private static <T, C extends PalettedContainerRO<T>> Codec<C> codec(
         Codec<T> elementCodec, Strategy<T> strategy, T defaultValue, PalettedContainerRO.Unpacker<T, C> unpacker
     ) {
-        return RecordCodecBuilder.<PalettedContainerRO.PackedData>create(
+        return RecordCodecBuilder.<PalettedContainerRO.PackedData<T>>create(
                 i -> i.group(
                         elementCodec.mapResult(ExtraCodecs.orElsePartial(defaultValue))
                             .listOf()
