--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -81,7 +_,7 @@
     }
 
     public <T extends Entity> EntityRenderer<? super T, ?> getRenderer(final T entity) {
-        return switch (entity) {
+        return (EntityRenderer<? super T, ?>) switch (entity) {
             case AbstractClientPlayer player: {
                 yield getAvatarRenderer(playerRenderers, player);
             }
