--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -93,9 +_,9 @@
     }
 
     public <T extends Entity> EntityRenderer<? super T, ?> getRenderer(T entity) {
-        return (EntityRenderer<? super T, ?>)(switch (entity) {
-            case AbstractClientPlayer player -> this.getAvatarRenderer((Map<PlayerModelType, AvatarRenderer<T>>)this.playerRenderers, (T)player);
-            case ClientMannequin mannequin -> this.getAvatarRenderer((Map<PlayerModelType, AvatarRenderer<T>>)this.mannequinRenderers, (T)mannequin);
+        return (switch (entity) {
+            case AbstractClientPlayer player -> this.getAvatarRenderer((Map)this.playerRenderers, player);
+            case ClientMannequin mannequin -> this.getAvatarRenderer((Map)this.mannequinRenderers, mannequin);
             default -> (EntityRenderer)this.renderers.get(entity.getType());
         });
     }
