--- a/net/minecraft/gametest/framework/GameTestServer.java
+++ b/net/minecraft/gametest/framework/GameTestServer.java
@@ -147,7 +147,7 @@
 
     @Override
     public boolean initServer() {
-        this.setPlayerList(new PlayerList(this, this, this.registries(), this.playerDataStorage, 1) {
+        this.setPlayerList(new PlayerList(this, this.registries(), this.playerDataStorage, 1) {
         });
         this.loadLevel();
         ServerLevel serverlevel = this.overworld();
