--- a/net/minecraft/client/gui/Font.java
+++ b/net/minecraft/client/gui/Font.java
@@ -138,7 +138,7 @@
     ) {
         int i = adjustColor(p_168650_);
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
+            p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
         );
 
         for (int j = -1; j <= 1; j++) {
@@ -161,7 +161,7 @@
         }
 
         Font.StringRenderOutput font$stringrenderoutput1 = new Font.StringRenderOutput(
-            this, p_168652_, p_168647_, p_168648_, adjustColor(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
+            p_168652_, p_168647_, p_168648_, adjustColor(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
         );
         p_168646_.accept(font$stringrenderoutput1);
         font$stringrenderoutput1.finish();
@@ -235,7 +235,7 @@
         int p_273199_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_
+            p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_
         );
         StringDecomposer.iterateFormatted(p_273765_, Style.EMPTY, font$stringrenderoutput);
         return font$stringrenderoutput.finish();
@@ -254,7 +254,7 @@
         int p_273371_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_
+            p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_
         );
         p_273322_.accept(font$stringrenderoutput);
         return font$stringrenderoutput.finish();
@@ -351,7 +351,6 @@
         }
 
         public StringRenderOutput(
-            Font param1,
             MultiBufferSource p_181365_,
             float p_181366_,
             float p_181367_,
@@ -361,11 +360,10 @@
             Font.DisplayMode p_181371_,
             int p_181372_
         ) {
-            this(var1, p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_);
+            this(p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_);
         }
 
         public StringRenderOutput(
-            Font param1,
             MultiBufferSource p_379354_,
             float p_380215_,
             float p_380204_,
@@ -376,7 +374,6 @@
             Font.DisplayMode p_380021_,
             int p_379625_
         ) {
-            this.this$0 = var1;
             this.bufferSource = p_379354_;
             this.x = p_380215_;
             this.y = p_380204_;
@@ -391,8 +388,8 @@
 
         @Override
         public boolean accept(int p_92967_, Style p_92968_, int p_92969_) {
-            FontSet fontset = this.this$0.getFontSet(p_92968_.getFont());
-            GlyphInfo glyphinfo = fontset.getGlyphInfo(p_92969_, this.this$0.filterFishyGlyphs);
+            FontSet fontset = Font.this.getFontSet(p_92968_.getFont());
+            GlyphInfo glyphinfo = fontset.getGlyphInfo(p_92969_, Font.this.filterFishyGlyphs);
             BakedGlyph bakedglyph = p_92968_.isObfuscated() && p_92969_ != 32 ? fontset.getRandomGlyph(glyphinfo) : fontset.getGlyph(p_92969_);
             boolean flag = p_92968_.isBold();
             TextColor textcolor = p_92968_.getColor();
@@ -401,7 +398,7 @@
             float f1 = p_92967_ == 0 ? this.x - 1.0F : this.x;
             if (this.backgroundColor != 0) {
                 BakedGlyph.Effect bakedglyph$effect = new BakedGlyph.Effect(f1, this.y + 9.0F, this.x + f, this.y - 1.0F, -0.01F, this.backgroundColor);
-                BakedGlyph bakedglyph1 = this.this$0.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
+                BakedGlyph bakedglyph1 = Font.this.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
                 VertexConsumer vertexconsumer = this.bufferSource.getBuffer(bakedglyph1.renderType(this.mode));
                 bakedglyph1.renderEffect(bakedglyph$effect, this.pose, vertexconsumer, this.packedLightCoords);
             }
@@ -410,7 +407,7 @@
                 float f2 = flag ? glyphinfo.getBoldOffset() : 0.0F;
                 float f4 = this.dropShadow ? glyphinfo.getShadowOffset() : 0.0F;
                 VertexConsumer vertexconsumer1 = this.bufferSource.getBuffer(bakedglyph.renderType(this.mode));
-                this.this$0
+                Font.this
                     .renderChar(bakedglyph, flag, p_92968_.isItalic(), f2, this.x + f4, this.y + f4, this.pose, vertexconsumer1, i, this.packedLightCoords);
             }
 
@@ -429,7 +426,7 @@
 
         public float finish() {
             if (this.effects != null) {
-                BakedGlyph bakedglyph = this.this$0.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
+                BakedGlyph bakedglyph = Font.this.getFontSet(Style.DEFAULT_FONT).whiteGlyph();
                 VertexConsumer vertexconsumer = this.bufferSource.getBuffer(bakedglyph.renderType(this.mode));
 
                 for (BakedGlyph.Effect bakedglyph$effect : this.effects) {
