--- a/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
@@ -111,14 +_,14 @@
             )
         );
         poseStack.popPose();
-        wnx = -beamGlowRadius;
+        float wnx_glow = -beamGlowRadius;
         float wnz = -beamGlowRadius;
-        enz = -beamGlowRadius;
-        wsx = -beamGlowRadius;
+        float enz_glow = -beamGlowRadius;
+        float wsx_glow = -beamGlowRadius;
         uu1 = 0.0F;
         uu2 = 1.0F;
-        vv2 = -1.0F + texVOff;
-        vv1 = height * scale + vv2;
+        float vv2_glow = -1.0F + texVOff;
+        float vv1_glow = height * scale + vv2_glow;
         submitNodeCollector.submitCustomGeometry(
             poseStack,
             RenderTypes.beaconBeam(beamLocation, true),
@@ -128,18 +_,18 @@
                 ARGB.color(32, color),
                 beamStart,
                 beamEnd,
-                wnx,
+                wnx_glow,
                 wnz,
                 beamGlowRadius,
-                enz,
-                wsx,
+                enz_glow,
+                wsx_glow,
                 beamGlowRadius,
                 beamGlowRadius,
                 beamGlowRadius,
                 0.0F,
                 1.0F,
-                vv1,
-                vv2
+                vv1_glow,
+                vv2_glow
             )
         );
         poseStack.popPose();
