--- a/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
+++ b/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
@@ -143,8 +143,7 @@
         this.initialized = true;
         this.showPage(this.page);
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_DONE, p_329743_ -> this.onClose()).width(200).build());
-        this.layout.visitWidgets(p_329745_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.repositionElements();
     }
 
