--- a/net/minecraft/server/packs/linkfs/LinkFSProvider.java
+++ b/net/minecraft/server/packs/linkfs/LinkFSProvider.java
@@ -76,7 +76,7 @@
         if (pathcontents$directorycontents == null) {
             throw new NotDirectoryException(p_250116_.toString());
         } else {
-            return new DirectoryStream<Path>(this) {
+            return new DirectoryStream<Path>() {
                 @Override
                 public Iterator<Path> iterator() {
                     return pathcontents$directorycontents.children().values().stream().filter(p_250987_ -> {
@@ -85,7 +85,7 @@
                         } catch (IOException ioexception) {
                             throw new DirectoryIteratorException(ioexception);
                         }
-                    }).map(p_249891_ -> p_249891_).iterator();
+                    }).map(p_249891_ -> (Path)p_249891_).iterator();
                 }
 
                 @Override
