--- a/net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen.java
@@ -59,8 +59,7 @@
         );
         gridlayout$rowhelper.addChild(Button.builder(CommonComponents.GUI_PROCEED, p_252248_ -> this.callback.accept(true)).build());
         gridlayout$rowhelper.addChild(Button.builder(CommonComponents.GUI_BACK, p_250397_ -> this.callback.accept(false)).build());
-        this.layout.visitWidgets(p_269625_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
         this.layout.arrangeElements();
         this.repositionElements();
     }
