--- a/net/minecraft/client/multiplayer/p2p/client/SignalingServiceClient.java
+++ b/net/minecraft/client/multiplayer/p2p/client/SignalingServiceClient.java
@@ -135,7 +_,7 @@
                 ),
                 this.executor
             )
-            .thenApply(var0 -> null)
+            .thenApply(var0 -> (Void)null)
             .exceptionallyCompose(err -> {
                 if (err.getCause() instanceof JsonRpcException rpcErr) {
                     SignalingException mapped = SignalingErrorMapper.fromJsonRpc(toPlayerId, rpcErr);
