--- a/net/minecraft/world/entity/animal/Animal.java
+++ b/net/minecraft/world/entity/animal/Animal.java
@@ -195,7 +195,7 @@
 
     @Nullable
     public ServerPlayer getLoveCause() {
-        return EntityReference.get(this.loveCause, this.level()::getPlayerByUUID, ServerPlayer.class);
+        return EntityReference.get(this.loveCause, (net.minecraft.world.level.entity.UUIDLookup<Player>)this.level()::getPlayerByUUID, ServerPlayer.class);
     }
 
     public boolean isInLove() {
