--- a/net/minecraft/world/entity/animal/fox/Fox.java
+++ b/net/minecraft/world/entity/animal/fox/Fox.java
@@ -382,7 +_,7 @@
     protected void readAdditionalSaveData(final ValueInput input) {
         super.readAdditionalSaveData(input);
         clearTrusted();
-        input.<List<?>>read("Trusted", (Codec) TRUSTED_LIST_CODEC).orElse(List.of()).forEach(this::addTrustedEntity);
+        input.read("Trusted", TRUSTED_LIST_CODEC).orElse(List.of()).forEach(this::addTrustedEntity);
         setSleeping(input.getBooleanOr("Sleeping", false));
         setVariant(input.read("Type", Variant.CODEC).orElse(Variant.DEFAULT));
         setSitting(input.getBooleanOr("Sitting", false));
