--- a/net/minecraft/world/entity/ai/Brain.java
+++ b/net/minecraft/world/entity/ai/Brain.java
@@ -235,7 +_,7 @@
     }
 
     private static <U> void callVisitor(Brain.Visitor visitor, MemoryModuleType<U> memoryModuleType, MemorySlot<?> slot) {
-        slot.visit(memoryModuleType, visitor);
+        ((MemorySlot<U>)slot).visit(memoryModuleType, visitor);
     }
 
     public <U> boolean isMemoryValue(MemoryModuleType<U> memoryType, U value) {
