--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -1685,7 +_,7 @@
             Comparator<Entry<String>> compareByCount = Comparator.comparingInt(Entry::getIntValue);
             return countByType.object2IntEntrySet()
                 .stream()
-                .sorted(compareByCount.reversed())
+                .sorted(Comparator.<Entry<String>, Integer>comparing(Entry::getIntValue).reversed())
                 .limit(5L)
                 .map(ex -> (String)ex.getKey() + ":" + ex.getIntValue())
                 .collect(Collectors.joining(","));
