--- a/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java
+++ b/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java
@@ -71,6 +71,6 @@
     private net.minecraft.world.level.GameRules.Value<?> getRuleValue(String p_449204_) {
         net.minecraft.world.level.GameRules.Key<?> key = this.getRuleKey(p_449204_)
             .orElseThrow(() -> new InvalidParameterJsonRpcException("Game rule '" + p_449204_ + "' does not exist"));
-        return this.server.getGameRules().getRule((net.minecraft.world.level.GameRules.Key<net.minecraft.world.level.GameRules.Value<?>>)key);
+        return this.server.getGameRules().getRule(key);
     }
 }
