--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -14,7 +14,7 @@
 public class InsideBrownianWalk {
     public static BehaviorControl<PathfinderMob> create(float p_259775_) {
         return BehaviorBuilder.create(
-            p_258399_ -> p_258399_.<MemoryAccessor>group(p_258399_.absent(MemoryModuleType.WALK_TARGET))
+            p_258399_ -> p_258399_.group(p_258399_.absent(MemoryModuleType.WALK_TARGET))
                     .apply(
                         p_258399_,
                         p_258397_ -> (p_258393_, p_258394_, p_258395_) -> {
