--- a/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
+++ b/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
@@ -15,7 +15,7 @@
 public class LocateHidingPlace {
     public static OneShot<LivingEntity> create(int p_259202_, float p_259881_, int p_259982_) {
         return BehaviorBuilder.create(
-            p_258505_ -> p_258505_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258505_ -> p_258505_.group(
                         p_258505_.absent(MemoryModuleType.WALK_TARGET),
                         p_258505_.registered(MemoryModuleType.HOME),
                         p_258505_.registered(MemoryModuleType.HIDING_PLACE),
