--- a/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
+++ b/net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
@@ -10,7 +10,7 @@
 public class StopHoldingItemIfNoLongerAdmiring {
     public static BehaviorControl<Piglin> create() {
         return BehaviorBuilder.create(
-            p_259197_ -> p_259197_.<MemoryAccessor>group(p_259197_.absent(MemoryModuleType.ADMIRING_ITEM))
+            p_259197_ -> p_259197_.group(p_259197_.absent(MemoryModuleType.ADMIRING_ITEM))
                     .apply(p_259197_, p_259512_ -> (p_325831_, p_325832_, p_325833_) -> {
                             if (!p_325832_.getOffhandItem().isEmpty() && !p_325832_.getOffhandItem().is(Items.SHIELD)) {
                                 PiglinAi.stopHoldingOffHandItem(p_325832_, true);
