--- a/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
@@ -167,8 +_,8 @@
         private final List<FormattedCharSequence> label;
         protected final List<AbstractWidget> children;
 
-        public GameRuleEntry(List<FormattedCharSequence> tooltip, Component label) {
             Objects.requireNonNull(EditGameRulesScreen.this);
+        public GameRuleEntry(@Nullable List<FormattedCharSequence> tooltip, Component label) {
             super(tooltip);
             this.children = Lists.newArrayList();
             this.label = EditGameRulesScreen.this.minecraft.font.split(label, 175);
