--- a/net/minecraft/util/StringRepresentable.java
+++ b/net/minecraft/util/StringRepresentable.java
@@ -42,7 +42,7 @@
 
     static <T extends StringRepresentable> Function<String, T> createNameLookup(T[] p_304419_, Function<String, String> p_304658_) {
         if (p_304419_.length > 16) {
-            Map<String, T> map = Arrays.<StringRepresentable>stream(p_304419_)
+            Map<String, T> map = Arrays.stream(p_304419_)
                 .collect(Collectors.toMap(p_304335_ -> p_304658_.apply(p_304335_.getSerializedName()), p_304719_ -> p_304719_));
             return p_304332_ -> p_304332_ == null ? null : map.get(p_304332_);
         } else {
