--- a/net/minecraft/client/renderer/PostChain.java
+++ b/net/minecraft/client/renderer/PostChain.java
@@ -106,114 +_,15 @@
             PostChainConfig.Input var9 = input;
             byte var10 = 0;
 
-            while (true) {
-                switch (SwitchBootstraps.typeSwitch<"typeSwitch",PostChainConfig.TextureInput,PostChainConfig.TargetInput>(var9, var10)) {
-                    case 0:
-                        PostChainConfig.TextureInput var11 = (PostChainConfig.TextureInput)var9;
-                        PostChainConfig.TextureInput var53 = var11;
-
-                        try {
-                            var54 = var53.samplerName();
-                        } catch (Throwable var32) {
-                            throw new MatchException(var32.toString(), var32);
-                        }
-
-                        String var37 = var54;
-                        var53 = var11;
-
-                        try {
-                            var56 = var53.location();
-                        } catch (Throwable var31) {
-                            throw new MatchException(var31.toString(), var31);
-                        }
-
-                        Identifier var38 = var56;
-                        var53 = var11;
-
-                        try {
-                            var58 = var53.width();
-                        } catch (Throwable var30) {
-                            throw new MatchException(var30.toString(), var30);
-                        }
-
-                        int var39 = var58;
-                        if (true) {
-                            var53 = var11;
-
-                            try {
-                                var60 = var53.height();
-                            } catch (Throwable var29) {
-                                throw new MatchException(var29.toString(), var29);
-                            }
-
-                            int var40 = var60;
-                            if (true) {
-                                var53 = var11;
-
-                                try {
-                                    var62 = var53.bilinear();
-                                } catch (Throwable var28) {
-                                    throw new MatchException(var28.toString(), var28);
-                                }
-
-                                boolean var41 = var62;
-                                if (true) {
+            {
+                switch (var9) {
+                    case PostChainConfig.TextureInput(String var37, Identifier var38, int var39, int var40, boolean var41):
                                     AbstractTexture var42 = textureManager.getTexture(var38.withPath(path -> "textures/effect/" + path + ".png"));
                                     inputs.add(new PostPass.TextureInput(var37, var42, var39, var40, var41));
                                     continue label110;
-                                }
-                            }
-                        }
-
-                        var10 = 1;
-                        break;
-                    case 1:
-                        PostChainConfig.TargetInput texture = (PostChainConfig.TargetInput)var9;
-                        PostChainConfig.TargetInput var10000 = texture;
-
-                        try {
-                            var46 = var10000.samplerName();
-                        } catch (Throwable var27) {
-                            throw new MatchException(var27.toString(), var27);
-                        }
-
-                        String var22 = var46;
-                        var10000 = texture;
-
-                        try {
-                            var48 = var10000.targetId();
-                        } catch (Throwable var26) {
-                            throw new MatchException(var26.toString(), var26);
-                        }
-
-                        Identifier var43 = var48;
-                        var10000 = texture;
-
-                        try {
-                            var50 = var10000.useDepthBuffer();
-                        } catch (Throwable var25) {
-                            throw new MatchException(var25.toString(), var25);
-                        }
-
-                        boolean var44 = var50;
-                        if (true) {
-                            var10000 = texture;
-
-                            try {
-                                var52 = var10000.bilinear();
-                            } catch (Throwable var24) {
-                                throw new MatchException(var24.toString(), var24);
-                            }
-
-                            boolean var45 = var52;
-                            if (true) {
+                    case PostChainConfig.TargetInput(String var22, Identifier var43, boolean var44, boolean var45):
                                 inputs.add(new PostPass.TargetInput(var22, var43, var44, var45));
                                 continue label110;
-                            }
-                        }
-
-                        var10 = 2;
-                        break;
                     default:
                         throw new MatchException(null, null);
                 }
