--- a/net/minecraft/network/protocol/login/ServerboundLoginAcknowledgedPacket.java
+++ b/net/minecraft/network/protocol/login/ServerboundLoginAcknowledgedPacket.java
@@ -4,14 +4,11 @@
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.protocol.Packet;
 
-public final class ServerboundLoginAcknowledgedPacket extends Record implements Packet<ServerLoginPacketListener> {
+public record ServerboundLoginAcknowledgedPacket() implements Packet<ServerLoginPacketListener> {
    public ServerboundLoginAcknowledgedPacket(FriendlyByteBuf p_295418_) {
       this();
    }
 
-   public ServerboundLoginAcknowledgedPacket() {
-   }
-
    public void m_5779_(FriendlyByteBuf p_295478_) {
    }
 
@@ -22,16 +19,4 @@
    public ConnectionProtocol m_291754_() {
       return ConnectionProtocol.CONFIGURATION;
    }
-
-   public final String toString() {
-      return this.toString<invokedynamic>(this);
    }
-
-   public final int hashCode() {
-      return this.hashCode<invokedynamic>(this);
-   }
-
-   public final boolean equals(Object p_296081_) {
-      return this.equals<invokedynamic>(this, p_296081_);
-   }
-}
