--- a/net/minecraft/commands/execution/tasks/BuildContexts.java
+++ b/net/minecraft/commands/execution/tasks/BuildContexts.java
@@ -50,7 +_,8 @@
                         modifiers = modifiers.setForked();
                     }
                     RedirectModifier<T> modifier = contextToRun.getRedirectModifier();
-                    if (modifier instanceof CustomModifierExecutor<T> customModifierExecutor) {
+                    if (modifier instanceof CustomModifierExecutor<?>) {
+                        var customModifierExecutor = (CustomModifierExecutor<T>) modifier;
                         customModifierExecutor.apply(originalSource, currentSources, currentStage, modifiers, ExecutionControl.create(context, frame));
                         return;
                     }
@@ -83,13 +_,14 @@
         }
         if (currentSources.isEmpty()) {
             if (modifiers.isReturn()) {
-                context.queueNext(new CommandQueueEntry<>(frame, FallthroughTask.instance()));
+                context.queueNext(new CommandQueueEntry<>(frame, FallthroughTask.<T>instance()));
             }
             return;
         }
         CommandContext<T> executeContext = currentStage.getTopContext();
         Command<T> command = executeContext.getCommand();
-        if (command instanceof CustomCommandExecutor<T> customCommandExecutor) {
+        if (command instanceof CustomCommandExecutor<?>) {
+            var customCommandExecutor = (CustomCommandExecutor<T>) command;
             ExecutionControl<T> executionControl = ExecutionControl.create(context, frame);
             for (T executionSource : currentSources) {
                 customCommandExecutor.run(executionSource, currentStage, modifiers, executionControl);
