--- a/net/minecraft/world/entity/monster/creaking/CreakingAi.java
+++ b/net/minecraft/world/entity/monster/creaking/CreakingAi.java
@@ -49,7 +49,7 @@
     static void initCoreActivity(Brain<Creaking> p_379953_) {
         p_379953_.addActivity(Activity.CORE, 0, ImmutableList.of(new Swim<Creaking>(0.8F) {
             protected boolean checkExtraStartConditions(ServerLevel p_380301_, Creaking p_380034_) {
-                return p_380034_.canMove() && super.checkExtraStartConditions(p_380301_, (LivingEntity)p_380034_);
+                return p_380034_.canMove() && super.checkExtraStartConditions(p_380301_, p_380034_);
             }
         }, new LookAtTargetSink(45, 90), new MoveToTargetSink()));
     }
