--- a/net/minecraft/world/entity/animal/Sheep.java
+++ b/net/minecraft/world/entity/animal/Sheep.java
@@ -60,7 +60,7 @@
     private static final int EAT_ANIMATION_TICKS = 40;
     private static final EntityDataAccessor<Byte> DATA_WOOL_ID = SynchedEntityData.defineId(Sheep.class, EntityDataSerializers.BYTE);
     private static final Map<DyeColor, Integer> COLOR_BY_DYE = Maps.newEnumMap(
-        Arrays.stream(DyeColor.values()).collect(Collectors.toMap(p_29868_ -> (DyeColor)p_29868_, Sheep::createSheepColor))
+        Arrays.stream(DyeColor.values()).collect(Collectors.toMap(java.util.function.Function.identity(), Sheep::createSheepColor))
     );
     private int eatAnimationTick;
     private EatBlockGoal eatBlockGoal;
