--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -532,7 +532,7 @@
     }
 
     private static <S extends StateHolder<?, S>, T extends Comparable<T>> S setValueHelper(S p_394311_, Property<T> p_394352_, Object p_394525_) {
-        return p_394311_.setValue(p_394352_, (Comparable)p_394525_);
+        return p_394311_.setValue(p_394352_, (T)p_394525_);
     }
 
     @Deprecated
