--- a/net/minecraft/server/jsonrpc/api/Schema.java
+++ b/net/minecraft/server/jsonrpc/api/Schema.java
@@ -39,15 +39,15 @@
             "Schema",
             p_451634_ -> RecordCodecBuilder.create(
                 p_457376_ -> p_457376_.group(
-                        (App<Mu<? extends Schema<?>>, Optional<URI>>)ReferenceUtil.REFERENCE_CODEC.optionalFieldOf("$ref").forGetter(Schema::reference),
-                        (App<Mu<? extends Schema<?>>, List<String>>)ExtraCodecs.compactListCodec(Codec.STRING)
+                        ReferenceUtil.REFERENCE_CODEC.optionalFieldOf("$ref").forGetter(Schema::reference),
+                        ExtraCodecs.compactListCodec(Codec.STRING)
                             .optionalFieldOf("type", List.of())
                             .forGetter(Schema::type),
-                        (App<Mu<? extends Schema<?>>, Optional<? extends Schema<?>>>)p_451634_.optionalFieldOf("items").forGetter(Schema::items),
-                        (App<Mu<? extends Schema<?>>, Map<String, ? extends Schema<?>>>)Codec.unboundedMap(Codec.STRING, p_451634_)
+                        p_451634_.optionalFieldOf("items").forGetter(Schema::items),
+                        Codec.unboundedMap(Codec.STRING, p_451634_)
                             .optionalFieldOf("properties", Map.of())
                             .forGetter(Schema::properties),
-                        (App<Mu<? extends Schema<?>>, List<String>>)Codec.STRING.listOf().optionalFieldOf("enum", List.of()).forGetter(Schema::enumValues)
+                        Codec.STRING.listOf().optionalFieldOf("enum", List.of()).forGetter(Schema::enumValues)
                     )
                     .apply(p_457376_, (p_457377_, p_457378_, p_457379_, p_457380_, p_457381_) -> null)
             )
