--- a/com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen.java
@@ -37,8 +37,7 @@
         this.backupInfoList = this.layout.addToContents(new RealmsBackupInfoScreen.BackupInfoList(this.minecraft));
         this.layout.addToFooter(Button.builder(CommonComponents.GUI_BACK, p_299754_ -> this.onClose()).build());
         this.repositionElements();
-        this.layout.visitWidgets(p_299756_ -> {
-        });
+        this.layout.visitWidgets(this::addRenderableWidget);
     }
 
     @Override
