--- a/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
@@ -20,7 +20,7 @@
 public class YieldJobSite {
     public static BehaviorControl<Villager> create(float p_259768_) {
         return BehaviorBuilder.create(
-            p_258916_ -> p_258916_.<MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor, MemoryAccessor>group(
+            p_258916_ -> p_258916_.group(
                         p_258916_.present(MemoryModuleType.POTENTIAL_JOB_SITE),
                         p_258916_.absent(MemoryModuleType.JOB_SITE),
                         p_258916_.present(MemoryModuleType.NEAREST_LIVING_ENTITIES),
