--- a/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
@@ -264,8 +264,8 @@
                                     } else if (this.cancelled) {
                                         this.uploadCancelled();
                                     } else {
-                                        j = (int)(new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
-                                        file1 = this.tarGzipArchive(new File(j, this.selectedLevel.getLevelId()));
+                                        File file = (new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
+                                        file1 = this.tarGzipArchive(new File(file, this.selectedLevel.getLevelId()));
                                         if (this.cancelled) {
                                             this.uploadCancelled();
                                         } else if (this.verify(file1)) {
@@ -330,13 +330,10 @@
                                 }
                             }
                         } catch (IOException ioexception) {
-                            uploadinfo = ioexception;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", ioexception.getMessage()));
                         } catch (RealmsServiceException realmsserviceexception) {
-                            uploadinfo = realmsserviceexception;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", realmsserviceexception.realmsError.errorMessage()));
                         } catch (InterruptedException interruptedexception1) {
-                            uploadinfo = interruptedexception1;
                             LOGGER.error("Could not acquire upload lock");
                         }
                     } finally {
