--- a/net/minecraft/client/StringSplitter.java
+++ b/net/minecraft/client/StringSplitter.java
@@ -118,7 +118,7 @@
 
     public FormattedText headByWidth(FormattedText p_92390_, int p_92391_, Style p_92392_) {
         final StringSplitter.WidthLimitedCharSink stringsplitter$widthlimitedcharsink = new StringSplitter.WidthLimitedCharSink((float)p_92391_);
-        return p_92390_.visit(new FormattedText.StyledContentConsumer<FormattedText>(this) {
+        return p_92390_.visit(new FormattedText.StyledContentConsumer<FormattedText>() {
             private final ComponentCollector collector = new ComponentCollector();
 
             @Override
